<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.40">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Coverage by climatic zones – Code Resources for MEC-24-1288</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./geocov_figs2.html" rel="next">
<link href="./building_bold.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-549806ee2085284f45b00abea8c6df48.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-626149efe8f5d16e1d391ba177679bf0.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./geocov_figs1.html">Geographic coverage</a></li><li class="breadcrumb-item"><a href="./geocov_figs1.html"><span class="chapter-title">Coverage by climatic zones</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Code Resources for MEC-24-1288</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Context</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introduction</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Building global BOLD data</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./building_bold.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Building the BOLD dataset</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Geographic coverage</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./geocov_figs1.html" class="sidebar-item-text sidebar-link active"><span class="chapter-title">Coverage by climatic zones</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./geocov_figs2.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Coverage by country</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">Taxonomic coverage</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./fetch_itis.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Fetching taxonomic data from ITIS</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./taxonomic_coverage.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Taxonomic Coverage</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true">
 <span class="menu-text">Case-study – Yellowstone</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./fetch_gbif.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Fetching plant occurrence records from GBIF</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./build_geocov_map.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Mapping global coverage from local sampling</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#figure-1.-read-in-combtab-dataset-of-global-barcode-data-from-bold-and-summarize." id="toc-figure-1.-read-in-combtab-dataset-of-global-barcode-data-from-bold-and-summarize." class="nav-link active" data-scroll-target="#figure-1.-read-in-combtab-dataset-of-global-barcode-data-from-bold-and-summarize.">Figure 1. Read in combtab dataset of global barcode data from BOLD and summarize.</a>
  <ul class="collapse">
  <li><a href="#assess-project-contibutions." id="toc-assess-project-contibutions." class="nav-link" data-scroll-target="#assess-project-contibutions.">Assess Project contibutions.</a></li>
  </ul></li>
  <li><a href="#figure-1b-c.-plots-for-number-of-specimens-across-lat-and-lon-for-each-barcode" id="toc-figure-1b-c.-plots-for-number-of-specimens-across-lat-and-lon-for-each-barcode" class="nav-link" data-scroll-target="#figure-1b-c.-plots-for-number-of-specimens-across-lat-and-lon-for-each-barcode">Figure 1B-C. Plots for number of specimens across lat and lon for each barcode</a></li>
  <li><a href="#figure-1d.-plot-mat-and-map-points-with-95-confidence-intervals" id="toc-figure-1d.-plot-mat-and-map-points-with-95-confidence-intervals" class="nav-link" data-scroll-target="#figure-1d.-plot-mat-and-map-points-with-95-confidence-intervals">Figure 1D. Plot MAT and MAP points with 95% confidence intervals</a></li>
  <li><a href="#figure-1a.-map-geography-of-plant-barcodes." id="toc-figure-1a.-map-geography-of-plant-barcodes." class="nav-link" data-scroll-target="#figure-1a.-map-geography-of-plant-barcodes.">Figure 1A. Map: Geography of plant barcodes.</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./geocov_figs1.html">Geographic coverage</a></li><li class="breadcrumb-item"><a href="./geocov_figs1.html"><span class="chapter-title">Coverage by climatic zones</span></a></li></ol></nav>
<div class="quarto-title">
<h1 class="title"><span class="chapter-title">Coverage by climatic zones</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>This notebook will generate plots of geographic coverage of barcodes available on BOLD, in relation to climatic zones, and for several genetic markers. It also assesses contributions to BOLD by Project ID.</p>
<p>We trim dataframe outputs to the first two rows for demonstartion, but full dataframes will be generated and stored in the R environment when running the code from source.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="do">## First check for the required packages, install if needed, and load the libraries.</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"BiocManager"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"sangerseqR"</span>)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>remotes<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">"ropensci/bold"</span>)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"pacman"</span>)) <span class="fu">install.packages</span>(<span class="st">"pacman"</span>)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(taxize, maps, ggplot2, countrycode, rgbif, data.table, raster, maps, mapproj, sf)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="figure-1.-read-in-combtab-dataset-of-global-barcode-data-from-bold-and-summarize." class="level1">
<h1>Figure 1. Read in combtab dataset of global barcode data from BOLD and summarize.</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>combtab <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../data/BoldPhyla_to_Families_combtab_v4.csv"</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="co">#View(combtab_20240624)</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>combtab_2 <span class="ot">&lt;-</span> (<span class="fu">unique</span>(combtab<span class="sc">$</span>family_name))</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combtab_2, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Acanthaceae" "Achariaceae"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">unique</span>(combtab<span class="sc">$</span>family_name), <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Acanthaceae" "Achariaceae"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>combtab_subset_barcodes <span class="ot">&lt;-</span> combtab <span class="sc">%&gt;%</span> <span class="fu">subset</span>(multi <span class="sc">!=</span> <span class="st">"---"</span> <span class="sc">&amp;</span> multi <span class="sc">!=</span> <span class="st">""</span>)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combtab_subset_barcodes, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  X.1 X   processid               institution_storing phylum_taxID
1   1 1   API120-12 Sri Ramaswamy Memorial University           12
2   2 2 CANGI002-17      Museu Paraense Emilio Goeldi           12
    phylum_name class_taxID    class_name order_taxID order_name family_taxID
1 Magnoliophyta          41 Magnoliopsida      121216   Lamiales       148533
2 Magnoliophyta          41 Magnoliopsida      121216   Lamiales       148533
  family_name subfamily_taxID subfamily_name genus_taxID  genus_name
1 Acanthaceae              NA                     415894 Peristrophe
2 Acanthaceae          264310   Acanthoideae      148534     Ruellia
  species_taxID             species_name subspecies_taxID subspecies_name
1        494465 Peristrophe bicalyculata               NA                
2        632816          Ruellia inflata               NA                
  collectiondate_start collectiondate_end lat lon coord_source coord_accuracy
1                   NA                 NA  NA  NA                          NA
2                   NA                 NA  NA  NA                          NA
  elev country province_state region sector exactsite rbcL matK trnL ITS2
1   NA                                                rbcL               
2   NA  Brazil           Para                         rbcL matK       ITS
           multi gb_rbcL gb_matK gb_trnL gb_ITS
1        rbcL---                               
2 rbcL-matK--ITS                               </code></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>combtab_split_barcodes <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(combtab<span class="sc">$</span>multi, <span class="st">"-"</span>)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combtab_split_barcodes, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[[1]]
[1] "rbcL" ""     ""    

[[2]]
[1] "rbcL" "matK" ""     "ITS" </code></pre>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">unlist</span>(combtab_split_barcodes)[<span class="fu">which</span>(<span class="fu">unlist</span>(combtab_split_barcodes)<span class="sc">!=</span><span class="st">""</span>)], <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "rbcL" "rbcL"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">which</span>(<span class="fu">unlist</span>(combtab_split_barcodes)<span class="sc">!=</span><span class="st">""</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 372460</code></pre>
</div>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert blank cells to NA</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>combtab[combtab <span class="sc">==</span> <span class="st">""</span>] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Count the number of barcodes per specimen</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>combtab<span class="sc">$</span>barcode_count <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="sc">!</span><span class="fu">is.na</span>(combtab[, <span class="fu">c</span>(<span class="st">"rbcL"</span>, <span class="st">"matK"</span>, <span class="st">"trnL"</span>, <span class="st">"ITS2"</span>)]))</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the total number of barcodes across all rows</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>total_barcodes <span class="ot">&lt;-</span> <span class="fu">sum</span>(combtab<span class="sc">$</span>barcode_count)</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the results</span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a><span class="co">#View(combtab)</span></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>total_barcodes</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 373582</code></pre>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># write combtab</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>combtab <span class="ot">&lt;-</span> combtab </span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(combtab, <span class="st">"combtab.csv"</span>)</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="co"># write out combtab here for supplementary/supporting information </span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(combtab<span class="sc">$</span>species_name))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 102887</code></pre>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(combtab<span class="sc">$</span>country))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 208</code></pre>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(combtab<span class="sc">$</span>family_name))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 651</code></pre>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(combtab<span class="sc">$</span>family_name))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 651</code></pre>
</div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate % north of tropic of cancer, south of tropic of capricorn, and between the two </span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>combtab_north_of_tropic_of_cancer <span class="ot">&lt;-</span> combtab <span class="sc">%&gt;%</span> <span class="fu">subset</span>(lat <span class="sc">&gt;</span> <span class="fl">23.4</span>)</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combtab_north_of_tropic_of_cancer, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>    X.1   X   processid      institution_storing phylum_taxID   phylum_name
106 152 152 GBVU4228-13 Mined from GenBank, NCBI           12 Magnoliophyta
107 153 153 GBVX3812-15 Mined from GenBank, NCBI           12 Magnoliophyta
    class_taxID    class_name order_taxID order_name family_taxID family_name
106          41 Magnoliopsida      121216   Lamiales       148533 Acanthaceae
107          41 Magnoliopsida      121216   Lamiales       148533 Acanthaceae
    subfamily_taxID subfamily_name genus_taxID genus_name species_taxID
106              NA           &lt;NA&gt;      554651 Clarkeasia        554652
107              NA           &lt;NA&gt;      210661  Avicennia        641297
             species_name subspecies_taxID subspecies_name collectiondate_start
106 Clarkeasia parviflora               NA            &lt;NA&gt;                   NA
107 Avicennia officinalis               NA            &lt;NA&gt;                   NA
    collectiondate_end     lat     lon                      coord_source
106                 NA 28.3009 84.1339 Coordinates from country centroid
107                 NA 23.5863 81.1730 Coordinates from country centroid
    coord_accuracy elev country province_state   region sector exactsite rbcL
106             NA   NA   Nepal           &lt;NA&gt; Narayani   &lt;NA&gt;      &lt;NA&gt; rbcL
107             NA   NA   India           &lt;NA&gt;     &lt;NA&gt;   &lt;NA&gt;      &lt;NA&gt; &lt;NA&gt;
    matK trnL ITS2   multi gb_rbcL gb_matK gb_trnL gb_ITS barcode_count
106 &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; rbcL---    &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             1
107 matK &lt;NA&gt; &lt;NA&gt; -matK--    &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>combtab_south_of_tropic_of_capricorn <span class="ot">&lt;-</span> combtab <span class="sc">%&gt;%</span> <span class="fu">subset</span>(lat <span class="sc">&lt;</span> <span class="sc">-</span><span class="fl">23.4</span>)</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combtab_south_of_tropic_of_capricorn, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   X.1  X   processid
48  55 55 GBVU4382-13
52  60 60  KNPA505-09
                                                        institution_storing
48                                                 Mined from GenBank, NCBI
52 University of Johannesburg, Department of Botany and Plant Biotechnology
   phylum_taxID   phylum_name class_taxID    class_name order_taxID order_name
48           12 Magnoliophyta          41 Magnoliopsida      121216   Lamiales
52           12 Magnoliophyta          41 Magnoliopsida      121216   Lamiales
   family_taxID family_name subfamily_taxID subfamily_name genus_taxID
48       148533 Acanthaceae              NA           &lt;NA&gt;      554845
52       148533 Acanthaceae              NA           &lt;NA&gt;      205058
   genus_name species_taxID               species_name subspecies_taxID
48 Petalidium        554846   Petalidium oblongifolium               NA
52   Ruspolia        205059 Ruspolia hypocrateriformis               NA
   subspecies_name collectiondate_start collectiondate_end      lat     lon
48            &lt;NA&gt;                   NA                 NA -28.5536 24.7525
52            &lt;NA&gt;                   NA                 NA -23.9440 31.7250
                        coord_source coord_accuracy elev      country
48 Coordinates from country centroid             NA   NA South Africa
52                              &lt;NA&gt;             NA  208 South Africa
   province_state region               sector
48           &lt;NA&gt;   &lt;NA&gt;                 &lt;NA&gt;
52        Limpopo   &lt;NA&gt; Kruger National Park
                                            exactsite rbcL matK trnL ITS2
48                                               &lt;NA&gt; rbcL &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;
52 Kruger National Park, Nxanatseni region  North,  L rbcL matK &lt;NA&gt; &lt;NA&gt;
         multi  gb_rbcL  gb_matK gb_trnL gb_ITS barcode_count
48     rbcL--- FJ868916     &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             1
52 rbcL-matK-- KY632577 KY632577    &lt;NA&gt;   &lt;NA&gt;             2</code></pre>
</div>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>combtab_between_cancer_capricorn <span class="ot">&lt;-</span> combtab <span class="sc">%&gt;%</span> <span class="fu">subset</span>(lat <span class="sc">&gt;=</span> <span class="sc">-</span><span class="fl">23.4</span> <span class="sc">&amp;</span> lat <span class="sc">&lt;=</span> <span class="fl">23.4</span>)</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combtab_between_cancer_capricorn, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   X.1  X   processid            institution_storing phylum_taxID   phylum_name
49  56 56 GENG1220-15 Gujarat Biodiversity Gene Bank           12 Magnoliophyta
50  57 57 GENG1760-16 Gujarat Biodiversity Gene Bank           12 Magnoliophyta
   class_taxID    class_name order_taxID order_name family_taxID family_name
49          41 Magnoliopsida      121216   Lamiales       148533 Acanthaceae
50          41 Magnoliopsida      121216   Lamiales       148533 Acanthaceae
   subfamily_taxID  subfamily_name genus_taxID genus_name species_taxID
49          480072 Thunbergioideae      273749 Thunbergia        317700
50          264310    Acanthoideae      148534    Ruellia            NA
             species_name subspecies_taxID subspecies_name collectiondate_start
49 Thunbergia grandiflora               NA            &lt;NA&gt;                   NA
50                   &lt;NA&gt;               NA            &lt;NA&gt;                   NA
   collectiondate_end    lat    lon    coord_source coord_accuracy elev country
49                 NA 22.306 73.278 Garmin etrax 10             NA  130   India
50                 NA 21.615 70.505            &lt;NA&gt;             NA  532   India
   province_state   region sector exactsite rbcL matK trnL ITS2   multi
49        Gujarat Vodadara   &lt;NA&gt;      &lt;NA&gt; rbcL &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; rbcL---
50        Gujarat Junagadh   &lt;NA&gt;     Etava rbcL &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; rbcL---
    gb_rbcL gb_matK gb_trnL gb_ITS barcode_count
49 KX641599    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             1
50 KJ535203    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             1</code></pre>
</div>
</div>
<section id="assess-project-contibutions." class="level2">
<h2 class="anchored" data-anchor-id="assess-project-contibutions.">Assess Project contibutions.</h2>
<p>This block will assess the localized sampling projects’ contributions to the overall BOLD database.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># make container column </span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="co"># overall</span></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combtab[<span class="fu">grep</span>(<span class="st">"UHURU"</span>, combtab<span class="sc">$</span>processid),], <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   X.1  X   processid
59  78 78 UHURU149-14
60  82 82 UHURU394-14
                                                                             institution_storing
59 Smithsonian Institution, National Museum of Natural History, United States National Herbarium
60 Smithsonian Institution, National Museum of Natural History, United States National Herbarium
   phylum_taxID   phylum_name class_taxID    class_name order_taxID order_name
59           12 Magnoliophyta          41 Magnoliopsida      121216   Lamiales
60           12 Magnoliophyta          41 Magnoliopsida      121216   Lamiales
   family_taxID family_name subfamily_taxID subfamily_name genus_taxID
59       148533 Acanthaceae              NA           &lt;NA&gt;      148750
60       148533 Acanthaceae              NA           &lt;NA&gt;      306313
   genus_name species_taxID            species_name subspecies_taxID
59   Justicia        591068 Justicia diclipteroides               NA
60 Crossandra        659703     Crossandra massaica               NA
   subspecies_name collectiondate_start collectiondate_end    lat     lon
59            &lt;NA&gt;                   NA                 NA 0.4810 36.8740
60            &lt;NA&gt;                   NA                 NA 0.2956 36.8824
   coord_source coord_accuracy elev country province_state   region sector
59         &lt;NA&gt;             NA 1650   Kenya    Rift Valley Laikipia   &lt;NA&gt;
60         &lt;NA&gt;             NA 1650   Kenya    Rift Valley Laikipia   &lt;NA&gt;
               exactsite rbcL matK trnL ITS2           multi  gb_rbcL  gb_matK
59 Mpala Research Centre rbcL matK trnL &lt;NA&gt; rbcL-matK-trnL- KM255065 KM255065
60 Mpala Research Centre rbcL matK trnL &lt;NA&gt; rbcL-matK-trnL- JX517979 JX517979
    gb_trnL gb_ITS barcode_count
59 KM255065   &lt;NA&gt;             3
60 JX517979   &lt;NA&gt;             3</code></pre>
</div>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>combtab<span class="sc">$</span>container<span class="ot">&lt;-</span><span class="fu">gsub</span>(<span class="st">"[0-9.-]"</span>, <span class="st">""</span>, combtab<span class="sc">$</span>processid)</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(combtab<span class="sc">$</span>container))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 379</code></pre>
</div>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sort</span>(<span class="fu">table</span>(combtab<span class="sc">$</span>container), <span class="at">decreasing=</span>T) <span class="co">#PNG, YNPBP, UHURU</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
GBITS PGBMA  GBVX  GBVP   JWP  GBVA  GBVC  GBVU  GBVE  GBVS  GBVF  GBVW  GBVI 
70028 10890  8000  6694  6103  5270  4999  4989  4983  4983  4979  4959  4953 
 GBVM  GBVL  GBVG  GBVH  GBVK  GBVJ  GBVB  GBVD  GBVT  GBVR  SAFH  GBVO  GBVN 
 4939  4935  4920  4915  4898  4886  4870  4864  4854  4773  4271  3810  3668 
 GBVY BABRU POWNA  EBLF  GBVQ CPBOL  CRCZ   FCA  FBPL MHPAD   DPC   SDH UHURU 
 3629  3232  3174  3127  2863  2527  2450  2179  2143  2126  1780  1733  1720 
 GENG EDTOL  PNOR  KNPA VPSBC  FPUK PLOMA BRYCA IDRCG IPSUB TCAFR ABGCO MHPAF 
 1708  1581  1474  1339  1279  1240  1149  1125  1061   839   780   665   630 
MKPCH  PLID CANGI GRASS   PNG SALIX YNPBP  BATT CALAK FSEUS GRDMO  AUSG MHPAC 
  622   590   564   553   551   540   527   520   508   502   498   479   448 
  TRM MHPAA  TDEF  SCBI CANGA GCUDG PRTRE   KSR ALOAF   API FAUCT HOSAM   TSA 
  434   432   429   425   422   384   379   376   350   349   342   341   338 
 NNOR FPOON  KNPB  SBIO WABLK TRIOP   ALM ANGMD CSPPA VEMSH TMBRY NOVOA GRDTX 
  337   331   330   324   319   315   314   313   301   297   296   281   280 
  FGB  PHAK GCUBG ITVPL APOSA PSYAF MXBZI TADCR CAATB   PIM POWNB CYPAF DBFZP 
  272   260   256   254   243   233   226   221   214   213   211   205   202 
FYNBL   VPL RBTEP  FOND EGPAM MAPLT  PNCA HOLIN   POA BPTPS   ECU  ITVI MARQF 
  199   197   196   195   188   188   187   184   184   181   179   176   175 
TRIKU BRYBC  CYAF IASVF HAIDA BBYUK  MADA    HB MXBTX BRALT  SERC   CPL  CPPB 
  175   174   173   172   170   164   164   160   160   159   159   157   148 
PEPER   SAA  PLGE  BPNP GPAGA  KPLA  CROT ENDEM COBIN UGEAB  WREP MKTRT  FLSH 
  147   143   142   140   138   137   135   135   127   122   120   119   117 
MADPL BSCAT   RBG PLKEN CAREX MKCRA CONOP NECLE DBMPP SFFIP   YHT   LWT   WAG 
  111   108   100    99    94    93    92    92    91    89    89    87    85 
SASCH CREUA LEDCL LTRBN ZERGT  SLNM  OSCH  RUPL ORPOL LOPHZ POEUA RBPLT VPSEA 
   84    83    83    83    83    82    81    81    79    78    78    78    76 
PLCHA  CRBS PCABL BRPLT KIFNE PANFT PLTHU  BTAM  GRII MEXCO PLNOR VLUZH KIFRH 
   75    74    74    73    72    70    69    68    68    68    63    63    62 
 BFAA COBIM  DBZN MLTVP  MPIN PLCOC RINGV  AGOP PLALG   SAP   AGL ATRIP  FSSM 
   61    60    60    60    60    60    58    55    55    55    54    54    54 
PIPER  CERF  BAMP MIPDG ZPLPP   CSU DVHTF  DIDA   INB  SIDA SWFRG BRYUM  DBHI 
   53    51    50    50    49    48    47    46    46    46    46    44    44 
 CGII  MSTH JSUIS  HERB  IMDB KBGPP  NTDB KUDZU MKSAL CACRB IHGTI ITVRT TCHDC 
   43    43    42    41    40    40    40    39    39    38    36    36    35 
KIFMO   THY  TNAU BBIHP CINCH INDET  PDBA  DBTS FERNO  FMED  SYKO PLNIP FAGRB 
   34    34    34    33    32    31    31    30    29    29    29    28    27 
PLNIA LCBAR CACMK FFFCC MELIA   MBB PLWEL CRCBS PGLLP ACBRC AGMCV KIFHE   ATG 
   27    26    25    24    23    22    22    21    21    20    20    19    18 
  BMS FQAGM SMPRS  CRCB GTICO TFOAX CRCBT PCUBC RBGEV AGVMK BBOBC  USMF   SDP 
   18    18    18    17    17    15    14    14    14    13    13    13    12 
CACTB ITICS KIFAT MKOXY STRBS THYME ITVGA MHPAB NELKE PLCHU  PLNB PLPAR  PTOC 
   11    11    11    11    11    11    10    10    10    10    10    10    10 
 UADY AGVRB FAGMK  HRUC IMMEX  PMRB SABIO BBOTP LIVCA   MCT PCCMN PLTOR SEQLD 
   10     9     9     9     9     9     9     8     8     8     8     8     8 
 SPBL AUCAS CABAR RHEUM VASCB ALSEU BSRUG  DBTB  PHSN PLLAM   RBJ ABCBF ACCCO 
    8     7     7     7     7     6     6     6     6     6     6     5     5 
 BEER BHEPS CBIHP  DRAC IIPSA  MAHA  MPNE  MTSS NPOAF  PLON QUNAM RSUJO  SSAA 
    5     5     5     5     5     5     5     5     5     5     5     5     5 
  CBL CSPPB EMPWG  RICE   BUL COMPL   GRJ ITICZ MSHQU  MTAT PCYMX  RJAS  AMBC 
    4     4     4     4     3     3     3     3     3     3     3     3     2 
BCBSL  DCCC   GYM  HIMS  HRBC  KFRI LGRAS ORIGV PLPET PLSUD SEEWI  STRM SUSHI 
    2     2     2     2     2     2     2     2     2     2     2     2     2 
  AKD   APP   ASJ DBALT  DLFW DMCSI DRCLS EOCCF  GUTB   ISO LAARS LADEE LAHAR 
    1     1     1     1     1     1     1     1     1     1     1     1     1 
LAMSK MASMD MCAND MCBRT MCCAL MCGFB MCIII MCJNM MCMAN MCUII MCUKI MCVAD MCYMB 
    1     1     1     1     1     1     1     1     1     1     1     1     1 
 PCGM PLBRU PLMUS PLREN SDCCF SIDBG  SLAT   SLX  SPAP  SPAR   SRB THIPP  TRNT 
    1     1     1     1     1     1     1     1     1     1     1     1     1 
  UMS   WKA 
    1     1 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="co"># for trnL</span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(combtab<span class="sc">$</span>container[<span class="fu">which</span>(combtab<span class="sc">$</span>trnL<span class="sc">==</span><span class="st">"trnL"</span>)]))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 40</code></pre>
</div>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sort</span>(<span class="fu">table</span>(combtab<span class="sc">$</span>container[<span class="fu">which</span>(combtab<span class="sc">$</span>trnL<span class="sc">==</span><span class="st">"trnL"</span>)]), <span class="at">decreasing=</span>T) <span class="co">#PNG, YNPBP, UHURU</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
UHURU BRYCA YNPBP   PNG TMBRY BRYBC  MADA BRALT HAIDA   YHT PANFT  DBZN KIFRH 
 1598  1125   502   496   296   174   164   159   144    89    70    60    51 
EDTOL  MPIN  FOND BBIHP KIFMO SASCH  SAFH   SAA GTICO KIFAT SABIO IMMEX INDET 
   50    49    47    33    32    32    24    19    17    11     9     8     8 
LIVCA   MCT MIPDG   LWT CBIHP   CBL EMPWG   TSA APOSA MKPCH  NNOR VASCB  FBPL 
    8     8     7     6     5     4     4     4     3     2     2     2     1 
MKTRT 
    1 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co"># percent of trnL sequences originating from top 5</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">sort</span>(<span class="fu">table</span>(combtab<span class="sc">$</span>container[<span class="fu">which</span>(combtab<span class="sc">$</span>trnL<span class="sc">==</span><span class="st">"trnL"</span>)]), <span class="at">decreasing=</span>T)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>])<span class="sc">/</span><span class="fu">sum</span>(<span class="fu">sort</span>(<span class="fu">table</span>(combtab<span class="sc">$</span>container[<span class="fu">which</span>(combtab<span class="sc">$</span>trnL<span class="sc">==</span><span class="st">"trnL"</span>)]), <span class="at">decreasing=</span>T)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.7545079</code></pre>
</div>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>(<span class="fu">sort</span>(<span class="fu">table</span>(combtab<span class="sc">$</span>container[<span class="fu">which</span>(combtab<span class="sc">$</span>trnL<span class="sc">==</span><span class="st">"trnL"</span>)]), <span class="at">decreasing=</span>T)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>])<span class="sc">/</span><span class="fu">sum</span>(<span class="fu">sort</span>(<span class="fu">table</span>(combtab<span class="sc">$</span>container[<span class="fu">which</span>(combtab<span class="sc">$</span>trnL<span class="sc">==</span><span class="st">"trnL"</span>)]), <span class="at">decreasing=</span>T)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
     UHURU      BRYCA      YNPBP        PNG      TMBRY 
0.30015026 0.21130729 0.09429001 0.09316304 0.05559730 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># number of specimen records mined from genbank </span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>combtab_mined_from_genbank<span class="ot">&lt;-</span> <span class="fu">subset</span> (combtab, institution_storing <span class="sc">==</span> <span class="st">"Mined from GenBank, NCBI"</span>)</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combtab_mined_from_genbank, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  X.1  X     processid      institution_storing phylum_taxID   phylum_name
4  11 11 GBITS29777-21 Mined from GenBank, NCBI           12 Magnoliophyta
5  12 12 GBITS29782-21 Mined from GenBank, NCBI           12 Magnoliophyta
  class_taxID    class_name order_taxID order_name family_taxID family_name
4          41 Magnoliopsida      121216   Lamiales       148533 Acanthaceae
5          41 Magnoliopsida      121216   Lamiales       148533 Acanthaceae
  subfamily_taxID subfamily_name genus_taxID genus_name species_taxID
4              NA           &lt;NA&gt;      148601 Aphelandra        801590
5              NA           &lt;NA&gt;      148601 Aphelandra       1100819
            species_name subspecies_taxID subspecies_name collectiondate_start
4      Aphelandra hylaea               NA            &lt;NA&gt;                   NA
5 Aphelandra maximiliana               NA            &lt;NA&gt;                   NA
  collectiondate_end lat lon coord_source coord_accuracy elev country
4                 NA  NA  NA         &lt;NA&gt;             NA   NA    &lt;NA&gt;
5                 NA  NA  NA         &lt;NA&gt;             NA   NA    &lt;NA&gt;
  province_state region sector exactsite rbcL matK trnL ITS2  multi gb_rbcL
4           &lt;NA&gt;   &lt;NA&gt;   &lt;NA&gt;      &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;  ITS #NAME?    &lt;NA&gt;
5           &lt;NA&gt;   &lt;NA&gt;   &lt;NA&gt;      &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;  ITS #NAME?    &lt;NA&gt;
  gb_matK gb_trnL gb_ITS barcode_count container
4    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             1     GBITS
5    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             1     GBITS</code></pre>
</div>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="co"># number of specimens in combtab identified to species</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>combtab_IDd_to_species <span class="ot">&lt;-</span> <span class="fu">subset</span>(combtab, species_name <span class="sc">!=</span> <span class="st">""</span>)</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combtab_IDd_to_species, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  X.1 X   processid               institution_storing phylum_taxID
1   1 1   API120-12 Sri Ramaswamy Memorial University           12
2   2 2 CANGI002-17      Museu Paraense Emilio Goeldi           12
    phylum_name class_taxID    class_name order_taxID order_name family_taxID
1 Magnoliophyta          41 Magnoliopsida      121216   Lamiales       148533
2 Magnoliophyta          41 Magnoliopsida      121216   Lamiales       148533
  family_name subfamily_taxID subfamily_name genus_taxID  genus_name
1 Acanthaceae              NA           &lt;NA&gt;      415894 Peristrophe
2 Acanthaceae          264310   Acanthoideae      148534     Ruellia
  species_taxID             species_name subspecies_taxID subspecies_name
1        494465 Peristrophe bicalyculata               NA            &lt;NA&gt;
2        632816          Ruellia inflata               NA            &lt;NA&gt;
  collectiondate_start collectiondate_end lat lon coord_source coord_accuracy
1                   NA                 NA  NA  NA         &lt;NA&gt;             NA
2                   NA                 NA  NA  NA         &lt;NA&gt;             NA
  elev country province_state region sector exactsite rbcL matK trnL ITS2
1   NA    &lt;NA&gt;           &lt;NA&gt;   &lt;NA&gt;   &lt;NA&gt;      &lt;NA&gt; rbcL &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;
2   NA  Brazil           Para   &lt;NA&gt;   &lt;NA&gt;      &lt;NA&gt; rbcL matK &lt;NA&gt;  ITS
           multi gb_rbcL gb_matK gb_trnL gb_ITS barcode_count container
1        rbcL---    &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             1       API
2 rbcL-matK--ITS    &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             3     CANGI</code></pre>
</div>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co"># number of sequences mined from genbank </span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>combtab_mined_from_genbank<span class="ot">&lt;-</span> <span class="fu">subset</span> (combtab, institution_storing <span class="sc">==</span> <span class="st">"Mined from GenBank, NCBI"</span>)</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combtab_mined_from_genbank, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  X.1  X     processid      institution_storing phylum_taxID   phylum_name
4  11 11 GBITS29777-21 Mined from GenBank, NCBI           12 Magnoliophyta
5  12 12 GBITS29782-21 Mined from GenBank, NCBI           12 Magnoliophyta
  class_taxID    class_name order_taxID order_name family_taxID family_name
4          41 Magnoliopsida      121216   Lamiales       148533 Acanthaceae
5          41 Magnoliopsida      121216   Lamiales       148533 Acanthaceae
  subfamily_taxID subfamily_name genus_taxID genus_name species_taxID
4              NA           &lt;NA&gt;      148601 Aphelandra        801590
5              NA           &lt;NA&gt;      148601 Aphelandra       1100819
            species_name subspecies_taxID subspecies_name collectiondate_start
4      Aphelandra hylaea               NA            &lt;NA&gt;                   NA
5 Aphelandra maximiliana               NA            &lt;NA&gt;                   NA
  collectiondate_end lat lon coord_source coord_accuracy elev country
4                 NA  NA  NA         &lt;NA&gt;             NA   NA    &lt;NA&gt;
5                 NA  NA  NA         &lt;NA&gt;             NA   NA    &lt;NA&gt;
  province_state region sector exactsite rbcL matK trnL ITS2  multi gb_rbcL
4           &lt;NA&gt;   &lt;NA&gt;   &lt;NA&gt;      &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;  ITS #NAME?    &lt;NA&gt;
5           &lt;NA&gt;   &lt;NA&gt;   &lt;NA&gt;      &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;  ITS #NAME?    &lt;NA&gt;
  gb_matK gb_trnL gb_ITS barcode_count container
4    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             1     GBITS
5    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             1     GBITS</code></pre>
</div>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>combtab_mined_from_genbank[combtab_mined_from_genbank <span class="sc">==</span> <span class="st">""</span>] <span class="ot">&lt;-</span> <span class="cn">NA</span> <span class="co"># Convert blank cells to NA</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>combtab_mined_from_genbank<span class="sc">$</span>barcode_count_mined <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="sc">!</span><span class="fu">is.na</span>(combtab_mined_from_genbank[, <span class="fu">c</span>(<span class="st">"rbcL"</span>, <span class="st">"matK"</span>, <span class="st">"trnL"</span>, <span class="st">"ITS2"</span>)])) <span class="co"># Count the number of barcodes per specimen</span></span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>total_barcodes_mined <span class="ot">&lt;-</span> <span class="fu">sum</span>(combtab_mined_from_genbank<span class="sc">$</span>barcode_count_mined) <span class="co"># Calculate the total number of barcodes across all rows</span></span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a><span class="co">#View(combtab_mined_from_genbank)</span></span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a>total_barcodes_mined</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 199591</code></pre>
</div>
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="co"># number of specimens in combtab with rbcL barcode </span></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>combtab_rbcL_barcodes <span class="ot">&lt;-</span> <span class="fu">subset</span>(combtab, rbcL <span class="sc">!=</span> <span class="st">""</span>)</span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combtab_rbcL_barcodes, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  X.1 X   processid               institution_storing phylum_taxID
1   1 1   API120-12 Sri Ramaswamy Memorial University           12
2   2 2 CANGI002-17      Museu Paraense Emilio Goeldi           12
    phylum_name class_taxID    class_name order_taxID order_name family_taxID
1 Magnoliophyta          41 Magnoliopsida      121216   Lamiales       148533
2 Magnoliophyta          41 Magnoliopsida      121216   Lamiales       148533
  family_name subfamily_taxID subfamily_name genus_taxID  genus_name
1 Acanthaceae              NA           &lt;NA&gt;      415894 Peristrophe
2 Acanthaceae          264310   Acanthoideae      148534     Ruellia
  species_taxID             species_name subspecies_taxID subspecies_name
1        494465 Peristrophe bicalyculata               NA            &lt;NA&gt;
2        632816          Ruellia inflata               NA            &lt;NA&gt;
  collectiondate_start collectiondate_end lat lon coord_source coord_accuracy
1                   NA                 NA  NA  NA         &lt;NA&gt;             NA
2                   NA                 NA  NA  NA         &lt;NA&gt;             NA
  elev country province_state region sector exactsite rbcL matK trnL ITS2
1   NA    &lt;NA&gt;           &lt;NA&gt;   &lt;NA&gt;   &lt;NA&gt;      &lt;NA&gt; rbcL &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;
2   NA  Brazil           Para   &lt;NA&gt;   &lt;NA&gt;      &lt;NA&gt; rbcL matK &lt;NA&gt;  ITS
           multi gb_rbcL gb_matK gb_trnL gb_ITS barcode_count container
1        rbcL---    &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             1       API
2 rbcL-matK--ITS    &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             3     CANGI</code></pre>
</div>
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="co"># number of specimens in combtab with matK barcode </span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>combtab_matK_barcodes <span class="ot">&lt;-</span> <span class="fu">subset</span>(combtab, matK <span class="sc">!=</span> <span class="st">""</span>)</span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combtab_matK_barcodes, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  X.1 X   processid                       institution_storing phylum_taxID
2   2 2 CANGI002-17              Museu Paraense Emilio Goeldi           12
3   8 8 EBLF1456-22 Herbarium of South China Botanical Garden           12
    phylum_name class_taxID    class_name order_taxID order_name family_taxID
2 Magnoliophyta          41 Magnoliopsida      121216   Lamiales       148533
3 Magnoliophyta          41 Magnoliopsida      121216   Lamiales       148533
  family_name subfamily_taxID subfamily_name genus_taxID    genus_name
2 Acanthaceae          264310   Acanthoideae      148534       Ruellia
3 Acanthaceae              NA           &lt;NA&gt;     1142731 Phlogacanthus
  species_taxID           species_name subspecies_taxID subspecies_name
2        632816        Ruellia inflata               NA            &lt;NA&gt;
3       1142732 Phlogacanthus colaniae               NA            &lt;NA&gt;
  collectiondate_start collectiondate_end lat lon coord_source coord_accuracy
2                   NA                 NA  NA  NA         &lt;NA&gt;             NA
3                   NA                 NA  NA  NA         &lt;NA&gt;             NA
  elev country province_state        region sector
2   NA  Brazil           Para          &lt;NA&gt;   &lt;NA&gt;
3  220   China        Guangxi Chongzuo City   &lt;NA&gt;
                                           exactsite rbcL matK trnL ITS2
2                                               &lt;NA&gt; rbcL matK &lt;NA&gt;  ITS
3 Nonggang Protection Station-No.3 Boundary Tablet-L rbcL matK &lt;NA&gt;  ITS
           multi gb_rbcL gb_matK gb_trnL gb_ITS barcode_count container
2 rbcL-matK--ITS    &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             3     CANGI
3 rbcL-matK--ITS    &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             3      EBLF</code></pre>
</div>
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="co"># number of specimens in combtab with ITS barcode </span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>combtab_ITS2_barcodes <span class="ot">&lt;-</span> <span class="fu">subset</span>(combtab, ITS2 <span class="sc">!=</span> <span class="st">""</span>)</span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combtab_ITS2_barcodes, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  X.1 X   processid                       institution_storing phylum_taxID
2   2 2 CANGI002-17              Museu Paraense Emilio Goeldi           12
3   8 8 EBLF1456-22 Herbarium of South China Botanical Garden           12
    phylum_name class_taxID    class_name order_taxID order_name family_taxID
2 Magnoliophyta          41 Magnoliopsida      121216   Lamiales       148533
3 Magnoliophyta          41 Magnoliopsida      121216   Lamiales       148533
  family_name subfamily_taxID subfamily_name genus_taxID    genus_name
2 Acanthaceae          264310   Acanthoideae      148534       Ruellia
3 Acanthaceae              NA           &lt;NA&gt;     1142731 Phlogacanthus
  species_taxID           species_name subspecies_taxID subspecies_name
2        632816        Ruellia inflata               NA            &lt;NA&gt;
3       1142732 Phlogacanthus colaniae               NA            &lt;NA&gt;
  collectiondate_start collectiondate_end lat lon coord_source coord_accuracy
2                   NA                 NA  NA  NA         &lt;NA&gt;             NA
3                   NA                 NA  NA  NA         &lt;NA&gt;             NA
  elev country province_state        region sector
2   NA  Brazil           Para          &lt;NA&gt;   &lt;NA&gt;
3  220   China        Guangxi Chongzuo City   &lt;NA&gt;
                                           exactsite rbcL matK trnL ITS2
2                                               &lt;NA&gt; rbcL matK &lt;NA&gt;  ITS
3 Nonggang Protection Station-No.3 Boundary Tablet-L rbcL matK &lt;NA&gt;  ITS
           multi gb_rbcL gb_matK gb_trnL gb_ITS barcode_count container
2 rbcL-matK--ITS    &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             3     CANGI
3 rbcL-matK--ITS    &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             3      EBLF</code></pre>
</div>
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="co"># number of specimens in combtab with trnL barcode </span></span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>combtab_trnL_barcodes <span class="ot">&lt;-</span> <span class="fu">subset</span>(combtab, trnL <span class="sc">!=</span> <span class="st">""</span>)</span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combtab_trnL_barcodes, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   X.1  X   processid
59  78 78 UHURU149-14
60  82 82 UHURU394-14
                                                                             institution_storing
59 Smithsonian Institution, National Museum of Natural History, United States National Herbarium
60 Smithsonian Institution, National Museum of Natural History, United States National Herbarium
   phylum_taxID   phylum_name class_taxID    class_name order_taxID order_name
59           12 Magnoliophyta          41 Magnoliopsida      121216   Lamiales
60           12 Magnoliophyta          41 Magnoliopsida      121216   Lamiales
   family_taxID family_name subfamily_taxID subfamily_name genus_taxID
59       148533 Acanthaceae              NA           &lt;NA&gt;      148750
60       148533 Acanthaceae              NA           &lt;NA&gt;      306313
   genus_name species_taxID            species_name subspecies_taxID
59   Justicia        591068 Justicia diclipteroides               NA
60 Crossandra        659703     Crossandra massaica               NA
   subspecies_name collectiondate_start collectiondate_end    lat     lon
59            &lt;NA&gt;                   NA                 NA 0.4810 36.8740
60            &lt;NA&gt;                   NA                 NA 0.2956 36.8824
   coord_source coord_accuracy elev country province_state   region sector
59         &lt;NA&gt;             NA 1650   Kenya    Rift Valley Laikipia   &lt;NA&gt;
60         &lt;NA&gt;             NA 1650   Kenya    Rift Valley Laikipia   &lt;NA&gt;
               exactsite rbcL matK trnL ITS2           multi  gb_rbcL  gb_matK
59 Mpala Research Centre rbcL matK trnL &lt;NA&gt; rbcL-matK-trnL- KM255065 KM255065
60 Mpala Research Centre rbcL matK trnL &lt;NA&gt; rbcL-matK-trnL- JX517979 JX517979
    gb_trnL gb_ITS barcode_count container
59 KM255065   &lt;NA&gt;             3     UHURU
60 JX517979   &lt;NA&gt;             3     UHURU</code></pre>
</div>
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="co"># number of genbank accession numbers in combtab </span></span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>combtab_for_genbank <span class="ot">&lt;-</span> combtab</span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>combtab_for_genbank[combtab_for_genbank <span class="sc">==</span> <span class="st">""</span>] <span class="ot">&lt;-</span> <span class="cn">NA</span> <span class="co"># Convert blank cells to NA</span></span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Count the number of barcodes per specimen (number of non-NA values per row in the specified columns)</span></span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a>combtab_for_genbank<span class="sc">$</span>accession_count <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="sc">!</span><span class="fu">is.na</span>(combtab_for_genbank[, <span class="fu">c</span>(<span class="st">"gb_rbcL"</span>, <span class="st">"gb_matK"</span>, <span class="st">"gb_trnL"</span>, <span class="st">"gb_ITS"</span>)]))</span>
<span id="cb60-7"><a href="#cb60-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-8"><a href="#cb60-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the total number of barcodes across all rows (sum of accession_count)</span></span>
<span id="cb60-9"><a href="#cb60-9" aria-hidden="true" tabindex="-1"></a>total_accessions <span class="ot">&lt;-</span> <span class="fu">sum</span>(combtab_for_genbank<span class="sc">$</span>accession_count)</span>
<span id="cb60-10"><a href="#cb60-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-11"><a href="#cb60-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the results</span></span>
<span id="cb60-12"><a href="#cb60-12" aria-hidden="true" tabindex="-1"></a><span class="co">#View(combtab_for_genbank)</span></span>
<span id="cb60-13"><a href="#cb60-13" aria-hidden="true" tabindex="-1"></a>total_accessions</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 262920</code></pre>
</div>
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="co"># number of specimens with country listed </span></span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>combtab_countries <span class="ot">&lt;-</span> <span class="fu">subset</span>(combtab, country <span class="sc">!=</span> <span class="st">""</span>)</span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combtab_countries, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  X.1 X   processid                       institution_storing phylum_taxID
2   2 2 CANGI002-17              Museu Paraense Emilio Goeldi           12
3   8 8 EBLF1456-22 Herbarium of South China Botanical Garden           12
    phylum_name class_taxID    class_name order_taxID order_name family_taxID
2 Magnoliophyta          41 Magnoliopsida      121216   Lamiales       148533
3 Magnoliophyta          41 Magnoliopsida      121216   Lamiales       148533
  family_name subfamily_taxID subfamily_name genus_taxID    genus_name
2 Acanthaceae          264310   Acanthoideae      148534       Ruellia
3 Acanthaceae              NA           &lt;NA&gt;     1142731 Phlogacanthus
  species_taxID           species_name subspecies_taxID subspecies_name
2        632816        Ruellia inflata               NA            &lt;NA&gt;
3       1142732 Phlogacanthus colaniae               NA            &lt;NA&gt;
  collectiondate_start collectiondate_end lat lon coord_source coord_accuracy
2                   NA                 NA  NA  NA         &lt;NA&gt;             NA
3                   NA                 NA  NA  NA         &lt;NA&gt;             NA
  elev country province_state        region sector
2   NA  Brazil           Para          &lt;NA&gt;   &lt;NA&gt;
3  220   China        Guangxi Chongzuo City   &lt;NA&gt;
                                           exactsite rbcL matK trnL ITS2
2                                               &lt;NA&gt; rbcL matK &lt;NA&gt;  ITS
3 Nonggang Protection Station-No.3 Boundary Tablet-L rbcL matK &lt;NA&gt;  ITS
           multi gb_rbcL gb_matK gb_trnL gb_ITS barcode_count container
2 rbcL-matK--ITS    &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             3     CANGI
3 rbcL-matK--ITS    &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             3      EBLF</code></pre>
</div>
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="co"># number of specimens with lat/long listed </span></span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a>combtab_lat <span class="ot">&lt;-</span> <span class="fu">subset</span>(combtab, lat <span class="sc">!=</span> <span class="st">""</span>)</span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combtab_lat, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   X.1  X   processid            institution_storing phylum_taxID   phylum_name
48  55 55 GBVU4382-13       Mined from GenBank, NCBI           12 Magnoliophyta
49  56 56 GENG1220-15 Gujarat Biodiversity Gene Bank           12 Magnoliophyta
   class_taxID    class_name order_taxID order_name family_taxID family_name
48          41 Magnoliopsida      121216   Lamiales       148533 Acanthaceae
49          41 Magnoliopsida      121216   Lamiales       148533 Acanthaceae
   subfamily_taxID  subfamily_name genus_taxID genus_name species_taxID
48              NA            &lt;NA&gt;      554845 Petalidium        554846
49          480072 Thunbergioideae      273749 Thunbergia        317700
               species_name subspecies_taxID subspecies_name
48 Petalidium oblongifolium               NA            &lt;NA&gt;
49   Thunbergia grandiflora               NA            &lt;NA&gt;
   collectiondate_start collectiondate_end      lat     lon
48                   NA                 NA -28.5536 24.7525
49                   NA                 NA  22.3060 73.2780
                        coord_source coord_accuracy elev      country
48 Coordinates from country centroid             NA   NA South Africa
49                   Garmin etrax 10             NA  130        India
   province_state   region sector exactsite rbcL matK trnL ITS2   multi
48           &lt;NA&gt;     &lt;NA&gt;   &lt;NA&gt;      &lt;NA&gt; rbcL &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; rbcL---
49        Gujarat Vodadara   &lt;NA&gt;      &lt;NA&gt; rbcL &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; rbcL---
    gb_rbcL gb_matK gb_trnL gb_ITS barcode_count container
48 FJ868916    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             1      GBVU
49 KX641599    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             1      GENG</code></pre>
</div>
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>combtab_lon <span class="ot">&lt;-</span> <span class="fu">subset</span>(combtab, lon <span class="sc">!=</span> <span class="st">""</span>)</span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combtab_lon, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   X.1  X   processid            institution_storing phylum_taxID   phylum_name
48  55 55 GBVU4382-13       Mined from GenBank, NCBI           12 Magnoliophyta
49  56 56 GENG1220-15 Gujarat Biodiversity Gene Bank           12 Magnoliophyta
   class_taxID    class_name order_taxID order_name family_taxID family_name
48          41 Magnoliopsida      121216   Lamiales       148533 Acanthaceae
49          41 Magnoliopsida      121216   Lamiales       148533 Acanthaceae
   subfamily_taxID  subfamily_name genus_taxID genus_name species_taxID
48              NA            &lt;NA&gt;      554845 Petalidium        554846
49          480072 Thunbergioideae      273749 Thunbergia        317700
               species_name subspecies_taxID subspecies_name
48 Petalidium oblongifolium               NA            &lt;NA&gt;
49   Thunbergia grandiflora               NA            &lt;NA&gt;
   collectiondate_start collectiondate_end      lat     lon
48                   NA                 NA -28.5536 24.7525
49                   NA                 NA  22.3060 73.2780
                        coord_source coord_accuracy elev      country
48 Coordinates from country centroid             NA   NA South Africa
49                   Garmin etrax 10             NA  130        India
   province_state   region sector exactsite rbcL matK trnL ITS2   multi
48           &lt;NA&gt;     &lt;NA&gt;   &lt;NA&gt;      &lt;NA&gt; rbcL &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; rbcL---
49        Gujarat Vodadara   &lt;NA&gt;      &lt;NA&gt; rbcL &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; rbcL---
    gb_rbcL gb_matK gb_trnL gb_ITS barcode_count container
48 FJ868916    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             1      GBVU
49 KX641599    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             1      GENG</code></pre>
</div>
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="co"># number of matK sequences</span></span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a>combtab_matK <span class="ot">&lt;-</span> <span class="fu">subset</span>(combtab, matK <span class="sc">==</span> <span class="st">"matK"</span>)</span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combtab_matK, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  X.1 X   processid                       institution_storing phylum_taxID
2   2 2 CANGI002-17              Museu Paraense Emilio Goeldi           12
3   8 8 EBLF1456-22 Herbarium of South China Botanical Garden           12
    phylum_name class_taxID    class_name order_taxID order_name family_taxID
2 Magnoliophyta          41 Magnoliopsida      121216   Lamiales       148533
3 Magnoliophyta          41 Magnoliopsida      121216   Lamiales       148533
  family_name subfamily_taxID subfamily_name genus_taxID    genus_name
2 Acanthaceae          264310   Acanthoideae      148534       Ruellia
3 Acanthaceae              NA           &lt;NA&gt;     1142731 Phlogacanthus
  species_taxID           species_name subspecies_taxID subspecies_name
2        632816        Ruellia inflata               NA            &lt;NA&gt;
3       1142732 Phlogacanthus colaniae               NA            &lt;NA&gt;
  collectiondate_start collectiondate_end lat lon coord_source coord_accuracy
2                   NA                 NA  NA  NA         &lt;NA&gt;             NA
3                   NA                 NA  NA  NA         &lt;NA&gt;             NA
  elev country province_state        region sector
2   NA  Brazil           Para          &lt;NA&gt;   &lt;NA&gt;
3  220   China        Guangxi Chongzuo City   &lt;NA&gt;
                                           exactsite rbcL matK trnL ITS2
2                                               &lt;NA&gt; rbcL matK &lt;NA&gt;  ITS
3 Nonggang Protection Station-No.3 Boundary Tablet-L rbcL matK &lt;NA&gt;  ITS
           multi gb_rbcL gb_matK gb_trnL gb_ITS barcode_count container
2 rbcL-matK--ITS    &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             3     CANGI
3 rbcL-matK--ITS    &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             3      EBLF</code></pre>
</div>
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="co"># number of matK sequences with lat/long data </span></span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a>combtab_matK_latlong <span class="ot">&lt;-</span> <span class="fu">subset</span>(combtab_matK, lat <span class="sc">!=</span> <span class="st">""</span>)</span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combtab_matK_latlong, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   X.1  X  processid
51  58 58 IMDB014-14
52  60 60 KNPA505-09
                                                        institution_storing
51             Birla Institute of Technology - Pilani, K.K.Birla Goa Campus
52 University of Johannesburg, Department of Botany and Plant Biotechnology
   phylum_taxID   phylum_name class_taxID    class_name order_taxID order_name
51           12 Magnoliophyta          41 Magnoliopsida      121216   Lamiales
52           12 Magnoliophyta          41 Magnoliopsida      121216   Lamiales
   family_taxID family_name subfamily_taxID subfamily_name genus_taxID
51       148533 Acanthaceae              NA           &lt;NA&gt;      390091
52       148533 Acanthaceae              NA           &lt;NA&gt;      205058
   genus_name species_taxID               species_name subspecies_taxID
51   Acanthus        641300       Acanthus ilicifolius               NA
52   Ruspolia        205059 Ruspolia hypocrateriformis               NA
   subspecies_name collectiondate_start collectiondate_end      lat     lon
51            &lt;NA&gt;                   NA                 NA  15.5256 73.8753
52            &lt;NA&gt;                   NA                 NA -23.9440 31.7250
   coord_source coord_accuracy elev      country province_state    region
51         &lt;NA&gt;             NA   NA        India            Goa North Goa
52         &lt;NA&gt;             NA  208 South Africa        Limpopo      &lt;NA&gt;
                 sector                                          exactsite rbcL
51      Mandovi estuary                                      Chorao Island rbcL
52 Kruger National Park Kruger National Park, Nxanatseni region  North,  L rbcL
   matK trnL ITS2       multi  gb_rbcL  gb_matK gb_trnL gb_ITS barcode_count
51 matK &lt;NA&gt; &lt;NA&gt; rbcL-matK-- DQ028462 DQ028462    &lt;NA&gt;   &lt;NA&gt;             2
52 matK &lt;NA&gt; &lt;NA&gt; rbcL-matK-- KY632577 KY632577    &lt;NA&gt;   &lt;NA&gt;             2
   container
51      IMDB
52      KNPA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="co"># number of rbcL sequences</span></span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a>combtab_rbcL <span class="ot">&lt;-</span> <span class="fu">subset</span>(combtab, rbcL <span class="sc">==</span> <span class="st">"rbcL"</span>)</span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combtab_rbcL, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  X.1 X   processid               institution_storing phylum_taxID
1   1 1   API120-12 Sri Ramaswamy Memorial University           12
2   2 2 CANGI002-17      Museu Paraense Emilio Goeldi           12
    phylum_name class_taxID    class_name order_taxID order_name family_taxID
1 Magnoliophyta          41 Magnoliopsida      121216   Lamiales       148533
2 Magnoliophyta          41 Magnoliopsida      121216   Lamiales       148533
  family_name subfamily_taxID subfamily_name genus_taxID  genus_name
1 Acanthaceae              NA           &lt;NA&gt;      415894 Peristrophe
2 Acanthaceae          264310   Acanthoideae      148534     Ruellia
  species_taxID             species_name subspecies_taxID subspecies_name
1        494465 Peristrophe bicalyculata               NA            &lt;NA&gt;
2        632816          Ruellia inflata               NA            &lt;NA&gt;
  collectiondate_start collectiondate_end lat lon coord_source coord_accuracy
1                   NA                 NA  NA  NA         &lt;NA&gt;             NA
2                   NA                 NA  NA  NA         &lt;NA&gt;             NA
  elev country province_state region sector exactsite rbcL matK trnL ITS2
1   NA    &lt;NA&gt;           &lt;NA&gt;   &lt;NA&gt;   &lt;NA&gt;      &lt;NA&gt; rbcL &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;
2   NA  Brazil           Para   &lt;NA&gt;   &lt;NA&gt;      &lt;NA&gt; rbcL matK &lt;NA&gt;  ITS
           multi gb_rbcL gb_matK gb_trnL gb_ITS barcode_count container
1        rbcL---    &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             1       API
2 rbcL-matK--ITS    &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             3     CANGI</code></pre>
</div>
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="co"># number of rbcL sequences with lat/long data </span></span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a>combtab_rbcL_latlong <span class="ot">&lt;-</span> <span class="fu">subset</span>(combtab_rbcL, lat <span class="sc">!=</span> <span class="st">""</span>)</span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combtab_rbcL_latlong, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   X.1  X   processid            institution_storing phylum_taxID   phylum_name
48  55 55 GBVU4382-13       Mined from GenBank, NCBI           12 Magnoliophyta
49  56 56 GENG1220-15 Gujarat Biodiversity Gene Bank           12 Magnoliophyta
   class_taxID    class_name order_taxID order_name family_taxID family_name
48          41 Magnoliopsida      121216   Lamiales       148533 Acanthaceae
49          41 Magnoliopsida      121216   Lamiales       148533 Acanthaceae
   subfamily_taxID  subfamily_name genus_taxID genus_name species_taxID
48              NA            &lt;NA&gt;      554845 Petalidium        554846
49          480072 Thunbergioideae      273749 Thunbergia        317700
               species_name subspecies_taxID subspecies_name
48 Petalidium oblongifolium               NA            &lt;NA&gt;
49   Thunbergia grandiflora               NA            &lt;NA&gt;
   collectiondate_start collectiondate_end      lat     lon
48                   NA                 NA -28.5536 24.7525
49                   NA                 NA  22.3060 73.2780
                        coord_source coord_accuracy elev      country
48 Coordinates from country centroid             NA   NA South Africa
49                   Garmin etrax 10             NA  130        India
   province_state   region sector exactsite rbcL matK trnL ITS2   multi
48           &lt;NA&gt;     &lt;NA&gt;   &lt;NA&gt;      &lt;NA&gt; rbcL &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; rbcL---
49        Gujarat Vodadara   &lt;NA&gt;      &lt;NA&gt; rbcL &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; rbcL---
    gb_rbcL gb_matK gb_trnL gb_ITS barcode_count container
48 FJ868916    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             1      GBVU
49 KX641599    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             1      GENG</code></pre>
</div>
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="co"># number of ITS sequences</span></span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a>combtab_ITS <span class="ot">&lt;-</span> <span class="fu">subset</span>(combtab, ITS2 <span class="sc">==</span> <span class="st">"ITS"</span>)</span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combtab_ITS, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  X.1 X   processid                       institution_storing phylum_taxID
2   2 2 CANGI002-17              Museu Paraense Emilio Goeldi           12
3   8 8 EBLF1456-22 Herbarium of South China Botanical Garden           12
    phylum_name class_taxID    class_name order_taxID order_name family_taxID
2 Magnoliophyta          41 Magnoliopsida      121216   Lamiales       148533
3 Magnoliophyta          41 Magnoliopsida      121216   Lamiales       148533
  family_name subfamily_taxID subfamily_name genus_taxID    genus_name
2 Acanthaceae          264310   Acanthoideae      148534       Ruellia
3 Acanthaceae              NA           &lt;NA&gt;     1142731 Phlogacanthus
  species_taxID           species_name subspecies_taxID subspecies_name
2        632816        Ruellia inflata               NA            &lt;NA&gt;
3       1142732 Phlogacanthus colaniae               NA            &lt;NA&gt;
  collectiondate_start collectiondate_end lat lon coord_source coord_accuracy
2                   NA                 NA  NA  NA         &lt;NA&gt;             NA
3                   NA                 NA  NA  NA         &lt;NA&gt;             NA
  elev country province_state        region sector
2   NA  Brazil           Para          &lt;NA&gt;   &lt;NA&gt;
3  220   China        Guangxi Chongzuo City   &lt;NA&gt;
                                           exactsite rbcL matK trnL ITS2
2                                               &lt;NA&gt; rbcL matK &lt;NA&gt;  ITS
3 Nonggang Protection Station-No.3 Boundary Tablet-L rbcL matK &lt;NA&gt;  ITS
           multi gb_rbcL gb_matK gb_trnL gb_ITS barcode_count container
2 rbcL-matK--ITS    &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             3     CANGI
3 rbcL-matK--ITS    &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             3      EBLF</code></pre>
</div>
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="co"># number of ITS sequences with lat/long data </span></span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a>combtab_ITS_latlong <span class="ot">&lt;-</span> <span class="fu">subset</span>(combtab_ITS, lat <span class="sc">!=</span> <span class="st">""</span>)</span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combtab_ITS_latlong, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   X.1  X    processid             institution_storing phylum_taxID
53  62 62 MHPAD3203-10 Area de Conservacion Guanacaste           12
54  65 65  MHPAD537-09 Area de Conservacion Guanacaste           12
     phylum_name class_taxID    class_name order_taxID order_name family_taxID
53 Magnoliophyta          41 Magnoliopsida      121216   Lamiales       148533
54 Magnoliophyta          41 Magnoliopsida      121216   Lamiales       148533
   family_name subfamily_taxID subfamily_name genus_taxID genus_name
53 Acanthaceae              NA           &lt;NA&gt;      148750   Justicia
54 Acanthaceae              NA           &lt;NA&gt;      148750   Justicia
   species_taxID        species_name subspecies_taxID subspecies_name
53        317738   Justicia brenesii               NA            &lt;NA&gt;
54        204815 Justicia arborecens               NA            &lt;NA&gt;
   collectiondate_start collectiondate_end    lat     lon coord_source
53                   NA                 NA 10.905 -85.279         &lt;NA&gt;
54                   NA                 NA 10.923 -85.465         &lt;NA&gt;
   coord_accuracy elev    country province_state
53             NA  410 Costa Rica       Alajuela
54             NA 1039 Costa Rica     Guanacaste
                            region            sector
53 Area de Conservacion Guanacaste Rincon Rainforest
54 Area de Conservacion Guanacaste      Sector Cacao
                                    exactsite rbcL matK trnL ITS2
53                             Sendero Rincon rbcL matK &lt;NA&gt;  ITS
54 Estacion Biologica Cacao, Sendero Arenales rbcL matK &lt;NA&gt;  ITS
            multi  gb_rbcL  gb_matK gb_trnL   gb_ITS barcode_count container
53 rbcL-matK--ITS AB114507 AB114507    &lt;NA&gt; AB114507             3     MHPAD
54 rbcL-matK--ITS AF188127 AF188127    &lt;NA&gt; AF188127             3     MHPAD</code></pre>
</div>
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="co"># number of trnL sequences</span></span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a>combtab_trnL <span class="ot">&lt;-</span> <span class="fu">subset</span>(combtab, trnL <span class="sc">==</span> <span class="st">"trnL"</span>)</span>
<span id="cb80-3"><a href="#cb80-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combtab_trnL, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   X.1  X   processid
59  78 78 UHURU149-14
60  82 82 UHURU394-14
                                                                             institution_storing
59 Smithsonian Institution, National Museum of Natural History, United States National Herbarium
60 Smithsonian Institution, National Museum of Natural History, United States National Herbarium
   phylum_taxID   phylum_name class_taxID    class_name order_taxID order_name
59           12 Magnoliophyta          41 Magnoliopsida      121216   Lamiales
60           12 Magnoliophyta          41 Magnoliopsida      121216   Lamiales
   family_taxID family_name subfamily_taxID subfamily_name genus_taxID
59       148533 Acanthaceae              NA           &lt;NA&gt;      148750
60       148533 Acanthaceae              NA           &lt;NA&gt;      306313
   genus_name species_taxID            species_name subspecies_taxID
59   Justicia        591068 Justicia diclipteroides               NA
60 Crossandra        659703     Crossandra massaica               NA
   subspecies_name collectiondate_start collectiondate_end    lat     lon
59            &lt;NA&gt;                   NA                 NA 0.4810 36.8740
60            &lt;NA&gt;                   NA                 NA 0.2956 36.8824
   coord_source coord_accuracy elev country province_state   region sector
59         &lt;NA&gt;             NA 1650   Kenya    Rift Valley Laikipia   &lt;NA&gt;
60         &lt;NA&gt;             NA 1650   Kenya    Rift Valley Laikipia   &lt;NA&gt;
               exactsite rbcL matK trnL ITS2           multi  gb_rbcL  gb_matK
59 Mpala Research Centre rbcL matK trnL &lt;NA&gt; rbcL-matK-trnL- KM255065 KM255065
60 Mpala Research Centre rbcL matK trnL &lt;NA&gt; rbcL-matK-trnL- JX517979 JX517979
    gb_trnL gb_ITS barcode_count container
59 KM255065   &lt;NA&gt;             3     UHURU
60 JX517979   &lt;NA&gt;             3     UHURU</code></pre>
</div>
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="co"># number of ITS sequences with lat/long data </span></span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>combtab_trnL_latlong <span class="ot">&lt;-</span> <span class="fu">subset</span>(combtab_trnL, lat <span class="sc">!=</span> <span class="st">""</span>)</span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combtab_trnL_latlong, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   X.1  X   processid
59  78 78 UHURU149-14
60  82 82 UHURU394-14
                                                                             institution_storing
59 Smithsonian Institution, National Museum of Natural History, United States National Herbarium
60 Smithsonian Institution, National Museum of Natural History, United States National Herbarium
   phylum_taxID   phylum_name class_taxID    class_name order_taxID order_name
59           12 Magnoliophyta          41 Magnoliopsida      121216   Lamiales
60           12 Magnoliophyta          41 Magnoliopsida      121216   Lamiales
   family_taxID family_name subfamily_taxID subfamily_name genus_taxID
59       148533 Acanthaceae              NA           &lt;NA&gt;      148750
60       148533 Acanthaceae              NA           &lt;NA&gt;      306313
   genus_name species_taxID            species_name subspecies_taxID
59   Justicia        591068 Justicia diclipteroides               NA
60 Crossandra        659703     Crossandra massaica               NA
   subspecies_name collectiondate_start collectiondate_end    lat     lon
59            &lt;NA&gt;                   NA                 NA 0.4810 36.8740
60            &lt;NA&gt;                   NA                 NA 0.2956 36.8824
   coord_source coord_accuracy elev country province_state   region sector
59         &lt;NA&gt;             NA 1650   Kenya    Rift Valley Laikipia   &lt;NA&gt;
60         &lt;NA&gt;             NA 1650   Kenya    Rift Valley Laikipia   &lt;NA&gt;
               exactsite rbcL matK trnL ITS2           multi  gb_rbcL  gb_matK
59 Mpala Research Centre rbcL matK trnL &lt;NA&gt; rbcL-matK-trnL- KM255065 KM255065
60 Mpala Research Centre rbcL matK trnL &lt;NA&gt; rbcL-matK-trnL- JX517979 JX517979
    gb_trnL gb_ITS barcode_count container
59 KM255065   &lt;NA&gt;             3     UHURU
60 JX517979   &lt;NA&gt;             3     UHURU</code></pre>
</div>
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="co"># number of overall sequences with lat/long data </span></span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a>combtab_latlong <span class="ot">&lt;-</span> <span class="fu">subset</span>(combtab, lat <span class="sc">!=</span> <span class="st">""</span>)</span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combtab_latlong, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   X.1  X   processid            institution_storing phylum_taxID   phylum_name
48  55 55 GBVU4382-13       Mined from GenBank, NCBI           12 Magnoliophyta
49  56 56 GENG1220-15 Gujarat Biodiversity Gene Bank           12 Magnoliophyta
   class_taxID    class_name order_taxID order_name family_taxID family_name
48          41 Magnoliopsida      121216   Lamiales       148533 Acanthaceae
49          41 Magnoliopsida      121216   Lamiales       148533 Acanthaceae
   subfamily_taxID  subfamily_name genus_taxID genus_name species_taxID
48              NA            &lt;NA&gt;      554845 Petalidium        554846
49          480072 Thunbergioideae      273749 Thunbergia        317700
               species_name subspecies_taxID subspecies_name
48 Petalidium oblongifolium               NA            &lt;NA&gt;
49   Thunbergia grandiflora               NA            &lt;NA&gt;
   collectiondate_start collectiondate_end      lat     lon
48                   NA                 NA -28.5536 24.7525
49                   NA                 NA  22.3060 73.2780
                        coord_source coord_accuracy elev      country
48 Coordinates from country centroid             NA   NA South Africa
49                   Garmin etrax 10             NA  130        India
   province_state   region sector exactsite rbcL matK trnL ITS2   multi
48           &lt;NA&gt;     &lt;NA&gt;   &lt;NA&gt;      &lt;NA&gt; rbcL &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; rbcL---
49        Gujarat Vodadara   &lt;NA&gt;      &lt;NA&gt; rbcL &lt;NA&gt; &lt;NA&gt; &lt;NA&gt; rbcL---
    gb_rbcL gb_matK gb_trnL gb_ITS barcode_count container
48 FJ868916    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             1      GBVU
49 KX641599    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             1      GENG</code></pre>
</div>
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(combtab<span class="sc">$</span>trnL)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
trnL 
5324 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(<span class="fu">factor</span>(combtab<span class="sc">$</span>trnL))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "trnL"</code></pre>
</div>
</div>
</section>
</section>
<section id="figure-1b-c.-plots-for-number-of-specimens-across-lat-and-lon-for-each-barcode" class="level1">
<h1>Figure 1B-C. Plots for number of specimens across lat and lon for each barcode</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="co"># round lat and lon to nearest one degree</span></span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a>latr <span class="ot">&lt;-</span><span class="fu">round</span>(combtab<span class="sc">$</span>lat)</span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a><span class="co">#latr</span></span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a>lonr <span class="ot">&lt;-</span> <span class="fu">round</span>(combtab<span class="sc">$</span>lon)</span>
<span id="cb90-5"><a href="#cb90-5" aria-hidden="true" tabindex="-1"></a><span class="co">#lonr</span></span>
<span id="cb90-6"><a href="#cb90-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-7"><a href="#cb90-7" aria-hidden="true" tabindex="-1"></a><span class="co"># create rounded data frame </span></span>
<span id="cb90-8"><a href="#cb90-8" aria-hidden="true" tabindex="-1"></a>combtab_rounded <span class="ot">&lt;-</span> combtab <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">mutate</span>(dplyr<span class="sc">::</span><span class="fu">across</span>(<span class="fu">c</span>(<span class="st">'lat'</span>, <span class="st">'lon'</span>), round, <span class="dv">0</span>))</span>
<span id="cb90-9"><a href="#cb90-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combtab_rounded, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  X.1 X   processid               institution_storing phylum_taxID
1   1 1   API120-12 Sri Ramaswamy Memorial University           12
2   2 2 CANGI002-17      Museu Paraense Emilio Goeldi           12
    phylum_name class_taxID    class_name order_taxID order_name family_taxID
1 Magnoliophyta          41 Magnoliopsida      121216   Lamiales       148533
2 Magnoliophyta          41 Magnoliopsida      121216   Lamiales       148533
  family_name subfamily_taxID subfamily_name genus_taxID  genus_name
1 Acanthaceae              NA           &lt;NA&gt;      415894 Peristrophe
2 Acanthaceae          264310   Acanthoideae      148534     Ruellia
  species_taxID             species_name subspecies_taxID subspecies_name
1        494465 Peristrophe bicalyculata               NA            &lt;NA&gt;
2        632816          Ruellia inflata               NA            &lt;NA&gt;
  collectiondate_start collectiondate_end lat lon coord_source coord_accuracy
1                   NA                 NA  NA  NA         &lt;NA&gt;             NA
2                   NA                 NA  NA  NA         &lt;NA&gt;             NA
  elev country province_state region sector exactsite rbcL matK trnL ITS2
1   NA    &lt;NA&gt;           &lt;NA&gt;   &lt;NA&gt;   &lt;NA&gt;      &lt;NA&gt; rbcL &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;
2   NA  Brazil           Para   &lt;NA&gt;   &lt;NA&gt;      &lt;NA&gt; rbcL matK &lt;NA&gt;  ITS
           multi gb_rbcL gb_matK gb_trnL gb_ITS barcode_count container
1        rbcL---    &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             1       API
2 rbcL-matK--ITS    &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             3     CANGI</code></pre>
</div>
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a>combtab_rounded_38 <span class="ot">&lt;-</span> <span class="fu">subset</span>(combtab_rounded, lat <span class="sc">==</span> <span class="dv">38</span>)</span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combtab_rounded_38, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>      X.1    X   processid      institution_storing phylum_taxID   phylum_name
366   507  507 GBVY3489-14 Mined from GenBank, NCBI           12 Magnoliophyta
1312 2051 2051 GBVU4183-13 Mined from GenBank, NCBI           12 Magnoliophyta
     class_taxID    class_name order_taxID order_name family_taxID family_name
366           41 Magnoliopsida      121216   Lamiales       148533 Acanthaceae
1312          41 Magnoliopsida      121216   Lamiales       148533 Acanthaceae
     subfamily_taxID  subfamily_name genus_taxID   genus_name species_taxID
366           480072 Thunbergioideae      273749   Thunbergia        601895
1312          264310    Acanthoideae      415904 Andrographis        415905
                species_name subspecies_taxID subspecies_name
366      Thunbergia coccinea               NA            &lt;NA&gt;
1312 Andrographis paniculata               NA            &lt;NA&gt;
     collectiondate_start collectiondate_end lat lon
366                    NA                 NA  38 105
1312                   NA                 NA  38 105
                          coord_source coord_accuracy elev country
366  Coordinates from country centroid             NA   NA   China
1312 Coordinates from country centroid             NA   NA   China
     province_state           region sector exactsite rbcL matK trnL ITS2
366            &lt;NA&gt; Yunnan, Mengsong   &lt;NA&gt;      &lt;NA&gt; rbcL &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;
1312           &lt;NA&gt;             &lt;NA&gt;   &lt;NA&gt;      &lt;NA&gt; rbcL &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;
       multi  gb_rbcL gb_matK gb_trnL gb_ITS barcode_count container
366  rbcL--- KR737446    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             1      GBVY
1312 rbcL---     &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             1      GBVU</code></pre>
</div>
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot longitude</span></span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a>marker_count_by_long_plot2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(combtab_rounded, <span class="fu">aes</span>(<span class="at">x =</span> lon)) <span class="sc">+</span></span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb94-4"><a href="#cb94-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"count"</span>, <span class="at">position =</span> <span class="st">"dodge"</span>, <span class="at">fill =</span> <span class="st">"gray"</span>) <span class="sc">+</span></span>
<span id="cb94-5"><a href="#cb94-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">180</span>, <span class="dv">180</span>, <span class="dv">60</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">180</span>, <span class="dv">180</span>)) <span class="sc">+</span> <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">5000</span>, <span class="dv">1000</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">5000</span>)) <span class="sc">+</span></span>
<span id="cb94-6"><a href="#cb94-6" aria-hidden="true" tabindex="-1"></a> <span class="fu">geom_line</span>(<span class="at">data =</span> <span class="fu">subset</span>(combtab_rounded, matK <span class="sc">==</span> <span class="st">"matK"</span>), <span class="fu">aes</span>(<span class="at">x =</span> lon), <span class="at">stat =</span> <span class="st">"count"</span>, <span class="at">col =</span> <span class="st">"#62BAAC"</span>, <span class="at">lwd =</span> <span class="fl">0.4</span>) <span class="sc">+</span></span>
<span id="cb94-7"><a href="#cb94-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> <span class="fu">subset</span>(combtab_rounded, rbcL <span class="sc">==</span> <span class="st">"rbcL"</span>), <span class="fu">aes</span>(<span class="at">x =</span> lon), <span class="at">stat =</span> <span class="st">"count"</span>, <span class="at">col =</span> <span class="st">"#EDA247"</span>, <span class="at">lwd =</span> <span class="fl">0.4</span>) <span class="sc">+</span></span>
<span id="cb94-8"><a href="#cb94-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> <span class="fu">subset</span>(combtab_rounded, trnL <span class="sc">==</span> <span class="st">"trnL"</span>), <span class="fu">aes</span>(<span class="at">x =</span> lon), <span class="at">stat =</span> <span class="st">"count"</span>, <span class="at">col =</span> <span class="st">"#006B5F"</span>, <span class="at">lwd =</span> <span class="fl">0.4</span>) <span class="sc">+</span></span>
<span id="cb94-9"><a href="#cb94-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> <span class="fu">subset</span>(combtab_rounded, ITS2 <span class="sc">==</span> <span class="st">"ITS"</span>), <span class="fu">aes</span>(<span class="at">x =</span> lon), <span class="at">stat =</span> <span class="st">"count"</span>, <span class="at">col =</span> <span class="st">"#703900"</span>, <span class="at">lwd =</span> <span class="fl">0.4</span>) <span class="sc">+</span></span>
<span id="cb94-10"><a href="#cb94-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">""</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">""</span>) <span class="co">#+ theme(panel.background = element_blank(), axis.text = element_blank())</span></span>
<span id="cb94-11"><a href="#cb94-11" aria-hidden="true" tabindex="-1"></a>marker_count_by_long_plot2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="geocov_figs1_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave("marker_count_by_long_plot2_20240624.pdf", marker_count_by_long_plot2, width = 13, height = 5, unit = "cm")</span></span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot latitude</span></span>
<span id="cb95-4"><a href="#cb95-4" aria-hidden="true" tabindex="-1"></a>marker_count_by_lat_plot_alllines <span class="ot">&lt;-</span> <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb95-5"><a href="#cb95-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb95-6"><a href="#cb95-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">90</span>, <span class="dv">90</span>, <span class="dv">45</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">90</span>, <span class="dv">90</span>)) <span class="sc">+</span>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">5500</span>, <span class="dv">2500</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">5500</span>)) <span class="sc">+</span></span>
<span id="cb95-7"><a href="#cb95-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_area</span>(<span class="at">data =</span> combtab_rounded, <span class="fu">aes</span>(<span class="at">x =</span> lat), <span class="at">stat =</span> <span class="st">"count"</span>, <span class="at">fill =</span> <span class="st">"lightgray"</span>, <span class="at">lwd =</span> <span class="fl">0.4</span>) <span class="sc">+</span> </span>
<span id="cb95-8"><a href="#cb95-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> <span class="fu">subset</span>(combtab_rounded, matK <span class="sc">==</span> <span class="st">"matK"</span>), <span class="fu">aes</span>(<span class="at">x =</span> lat), <span class="at">stat =</span> <span class="st">"count"</span>, <span class="at">col =</span> <span class="st">"#62BAAC"</span>, <span class="at">lwd =</span> <span class="fl">0.4</span>) <span class="sc">+</span></span>
<span id="cb95-9"><a href="#cb95-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> <span class="fu">subset</span>(combtab_rounded, rbcL <span class="sc">==</span> <span class="st">"rbcL"</span>), <span class="fu">aes</span>(<span class="at">x =</span> lat), <span class="at">stat =</span> <span class="st">"count"</span>, <span class="at">col =</span> <span class="st">"#EDA247"</span>, <span class="at">lwd =</span> <span class="fl">0.4</span>) <span class="sc">+</span></span>
<span id="cb95-10"><a href="#cb95-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> <span class="fu">subset</span>(combtab_rounded, trnL <span class="sc">==</span> <span class="st">"trnL"</span>), <span class="fu">aes</span>(<span class="at">x =</span> lat), <span class="at">stat =</span> <span class="st">"count"</span>, <span class="at">col =</span> <span class="st">"#006B5F"</span>, <span class="at">lwd =</span> <span class="fl">0.4</span>)<span class="sc">+</span> </span>
<span id="cb95-11"><a href="#cb95-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> <span class="fu">subset</span>(combtab_rounded, ITS2 <span class="sc">==</span> <span class="st">"ITS"</span>), <span class="fu">aes</span>(<span class="at">x =</span> lat), <span class="at">stat =</span> <span class="st">"count"</span>, <span class="at">col =</span> <span class="st">"#703900"</span>, <span class="at">lwd =</span> <span class="fl">0.4</span>)<span class="sc">+</span> </span>
<span id="cb95-12"><a href="#cb95-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">""</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">""</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_blank</span>(), <span class="at">axis.text =</span> <span class="fu">element_blank</span>())</span>
<span id="cb95-13"><a href="#cb95-13" aria-hidden="true" tabindex="-1"></a>marker_count_by_lat_plot_alllines</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="geocov_figs1_files/figure-html/unnamed-chunk-3-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave("marker_count_by_lat_plot_alllines_20240626_2.pdf", marker_count_by_lat_plot_alllines, width = 8.6, height = 5, unit = "cm")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a>combtab_matK <span class="ot">&lt;-</span> <span class="fu">sum</span>(combtab<span class="sc">$</span>matK <span class="sc">==</span> <span class="st">"matK"</span>)</span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a>combtab_rbcL <span class="ot">&lt;-</span> combtab <span class="sc">%&gt;%</span> <span class="fu">subset</span>(rbcL <span class="sc">==</span> <span class="st">"rbcL"</span>)</span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combtab_rbcL, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  X.1 X   processid               institution_storing phylum_taxID
1   1 1   API120-12 Sri Ramaswamy Memorial University           12
2   2 2 CANGI002-17      Museu Paraense Emilio Goeldi           12
    phylum_name class_taxID    class_name order_taxID order_name family_taxID
1 Magnoliophyta          41 Magnoliopsida      121216   Lamiales       148533
2 Magnoliophyta          41 Magnoliopsida      121216   Lamiales       148533
  family_name subfamily_taxID subfamily_name genus_taxID  genus_name
1 Acanthaceae              NA           &lt;NA&gt;      415894 Peristrophe
2 Acanthaceae          264310   Acanthoideae      148534     Ruellia
  species_taxID             species_name subspecies_taxID subspecies_name
1        494465 Peristrophe bicalyculata               NA            &lt;NA&gt;
2        632816          Ruellia inflata               NA            &lt;NA&gt;
  collectiondate_start collectiondate_end lat lon coord_source coord_accuracy
1                   NA                 NA  NA  NA         &lt;NA&gt;             NA
2                   NA                 NA  NA  NA         &lt;NA&gt;             NA
  elev country province_state region sector exactsite rbcL matK trnL ITS2
1   NA    &lt;NA&gt;           &lt;NA&gt;   &lt;NA&gt;   &lt;NA&gt;      &lt;NA&gt; rbcL &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;
2   NA  Brazil           Para   &lt;NA&gt;   &lt;NA&gt;      &lt;NA&gt; rbcL matK &lt;NA&gt;  ITS
           multi gb_rbcL gb_matK gb_trnL gb_ITS barcode_count container
1        rbcL---    &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             1       API
2 rbcL-matK--ITS    &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;             3     CANGI</code></pre>
</div>
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a>matK <span class="ot">&lt;-</span> <span class="fu">subset</span>(combtab_rounded, matK <span class="sc">==</span> <span class="st">"matK"</span>) </span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a>rbcL <span class="ot">&lt;-</span> <span class="fu">subset</span>(combtab_rounded, rbcL <span class="sc">==</span> <span class="st">"rbcL"</span>) </span>
<span id="cb99-3"><a href="#cb99-3" aria-hidden="true" tabindex="-1"></a>trnL <span class="ot">&lt;-</span> <span class="fu">subset</span>(combtab_rounded, trnL <span class="sc">==</span> <span class="st">"trnL"</span>) </span>
<span id="cb99-4"><a href="#cb99-4" aria-hidden="true" tabindex="-1"></a>ITS <span class="ot">&lt;-</span> <span class="fu">subset</span>(combtab_rounded, ITS2 <span class="sc">==</span> <span class="st">"ITS"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="figure-1d.-plot-mat-and-map-points-with-95-confidence-intervals" class="level1">
<h1>Figure 1D. Plot MAT and MAP points with 95% confidence intervals</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="do">#####</span></span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Worldclim connect lat lon specimen records (rounded to nearest degree and dereplicated) with temperature and precip</span></span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a><span class="do">#####</span></span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a><span class="co"># dereplicate sites rounded to nearest degree</span></span>
<span id="cb100-5"><a href="#cb100-5" aria-hidden="true" tabindex="-1"></a>sitepix<span class="ot">&lt;-</span><span class="fu">cbind</span>(lonr, latr)</span>
<span id="cb100-6"><a href="#cb100-6" aria-hidden="true" tabindex="-1"></a>sitepix<span class="ot">&lt;-</span><span class="fu">unique</span>(sitepix)</span>
<span id="cb100-7"><a href="#cb100-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-8"><a href="#cb100-8" aria-hidden="true" tabindex="-1"></a><span class="co"># load worldclim data</span></span>
<span id="cb100-9"><a href="#cb100-9" aria-hidden="true" tabindex="-1"></a>    <span class="co"># use 10 min for testing download directly from worldclim but here are some dropbox links: https://www.dropbox.com/scl/fo/9yl4dt2tli37qj24smitc/h?rlkey=z09vzcz1p6vg49pt48o92pyed&amp;dl=0</span></span>
<span id="cb100-10"><a href="#cb100-10" aria-hidden="true" tabindex="-1"></a>t.mean.files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"../data/wc2.1_10m_tavg/"</span>, <span class="st">".tif"</span>, <span class="at">full.names=</span><span class="cn">TRUE</span>)</span>
<span id="cb100-11"><a href="#cb100-11" aria-hidden="true" tabindex="-1"></a>p.mean.files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"../data/wc2.1_10m_prec/"</span>, <span class="st">".tif"</span>, <span class="at">full.names=</span><span class="cn">TRUE</span>)</span>
<span id="cb100-12"><a href="#cb100-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-13"><a href="#cb100-13" aria-hidden="true" tabindex="-1"></a><span class="co"># temperature (using the raster package)</span></span>
<span id="cb100-14"><a href="#cb100-14" aria-hidden="true" tabindex="-1"></a>t.mean <span class="ot">&lt;-</span> <span class="fu">stack</span>(t.mean.files)</span>
<span id="cb100-15"><a href="#cb100-15" aria-hidden="true" tabindex="-1"></a>t.mean <span class="ot">&lt;-</span><span class="fu">mean</span>(t.mean)</span>
<span id="cb100-16"><a href="#cb100-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-17"><a href="#cb100-17" aria-hidden="true" tabindex="-1"></a><span class="co"># make a 'samples' data frame to extract temp values from barcode localities</span></span>
<span id="cb100-18"><a href="#cb100-18" aria-hidden="true" tabindex="-1"></a>samples <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">processid=</span><span class="fu">seq</span>(<span class="dv">1</span>,<span class="fu">nrow</span>(sitepix)), <span class="at">lon=</span>sitepix[,<span class="dv">1</span>], <span class="at">lat=</span>sitepix[,<span class="dv">2</span>], <span class="at">row.names=</span><span class="st">"processid"</span>)</span>
<span id="cb100-19"><a href="#cb100-19" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(samples, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  lon lat
1  NA  NA
2  25 -29</code></pre>
</div>
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a>samples <span class="ot">&lt;-</span> samples[<span class="fu">which</span>(samples<span class="sc">$</span>lat<span class="sc">!=</span><span class="st">"NA"</span>),]</span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(samples, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  lon lat
2  25 -29
3  73  22</code></pre>
</div>
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract data from RasterLayer</span></span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a>temp.data <span class="ot">&lt;-</span> raster<span class="sc">::</span><span class="fu">extract</span>(t.mean, samples)</span>
<span id="cb104-3"><a href="#cb104-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(temp.data, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 17.8276 27.2780</code></pre>
</div>
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a>sampletempMeans <span class="ot">&lt;-</span> temp.data</span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">which</span>(<span class="fu">is.na</span>(sampletempMeans))) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 393</code></pre>
</div>
<div class="sourceCode cell-code" id="cb108"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="co"># precipitation</span></span>
<span id="cb108-2"><a href="#cb108-2" aria-hidden="true" tabindex="-1"></a>p.mean <span class="ot">&lt;-</span> <span class="fu">stack</span>(p.mean.files)</span>
<span id="cb108-3"><a href="#cb108-3" aria-hidden="true" tabindex="-1"></a>p.mean <span class="ot">&lt;-</span><span class="fu">sum</span>(p.mean)<span class="sc">/</span><span class="dv">10</span>     <span class="co"># divide by 10 to convert from mm to cm</span></span>
<span id="cb108-4"><a href="#cb108-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-5"><a href="#cb108-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract data from RasterLayer</span></span>
<span id="cb108-6"><a href="#cb108-6" aria-hidden="true" tabindex="-1"></a>precip.data <span class="ot">&lt;-</span> raster<span class="sc">::</span><span class="fu">extract</span>(p.mean, samples)</span>
<span id="cb108-7"><a href="#cb108-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(precip.data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1]  43.6  88.5  53.2 414.5  54.9 275.5</code></pre>
</div>
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a>sampleprecipMeans <span class="ot">&lt;-</span> precip.data</span>
<span id="cb110-2"><a href="#cb110-2" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">which</span>(<span class="fu">is.na</span>(sampleprecipMeans))) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 393</code></pre>
</div>
<div class="sourceCode cell-code" id="cb112"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot Figure 1d</span></span>
<span id="cb112-2"><a href="#cb112-2" aria-hidden="true" tabindex="-1"></a><span class="co"># get random points to plot as background</span></span>
<span id="cb112-3"><a href="#cb112-3" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">20140816</span>)</span>
<span id="cb112-4"><a href="#cb112-4" aria-hidden="true" tabindex="-1"></a>nrand<span class="ot">&lt;-</span><span class="dv">100000</span></span>
<span id="cb112-5"><a href="#cb112-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-6"><a href="#cb112-6" aria-hidden="true" tabindex="-1"></a>randomtemp <span class="ot">&lt;-</span> <span class="fu">sampleRandom</span>(<span class="fu">mean</span>(t.mean), nrand, <span class="at">xy=</span>T)</span>
<span id="cb112-7"><a href="#cb112-7" aria-hidden="true" tabindex="-1"></a>randomprecip <span class="ot">&lt;-</span> <span class="fu">extract</span>(<span class="fu">mean</span>(p.mean), randomtemp[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>])</span>
<span id="cb112-8"><a href="#cb112-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-9"><a href="#cb112-9" aria-hidden="true" tabindex="-1"></a>randomtempMeans <span class="ot">&lt;-</span> randomtemp[,<span class="dv">3</span>]</span>
<span id="cb112-10"><a href="#cb112-10" aria-hidden="true" tabindex="-1"></a>randomprecipMeans <span class="ot">&lt;-</span> randomprecip</span>
<span id="cb112-11"><a href="#cb112-11" aria-hidden="true" tabindex="-1"></a>randompoints <span class="ot">&lt;-</span> <span class="fu">cbind</span>(randomtempMeans, randomprecipMeans)</span>
<span id="cb112-12"><a href="#cb112-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-13"><a href="#cb112-13" aria-hidden="true" tabindex="-1"></a><span class="co"># find the upper / lower 95 percentile for temp and precip to plot based on BOLD samples</span></span>
<span id="cb112-14"><a href="#cb112-14" aria-hidden="true" tabindex="-1"></a>temp95 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(sampletempMeans, <span class="fu">c</span>(<span class="fl">0.05</span>), <span class="at">na.rm=</span>T)</span>
<span id="cb112-15"><a href="#cb112-15" aria-hidden="true" tabindex="-1"></a>precip95 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(sampleprecipMeans, <span class="fu">c</span>(<span class="fl">0.95</span>), <span class="at">na.rm=</span>T) </span>
<span id="cb112-16"><a href="#cb112-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-17"><a href="#cb112-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot figure 1D</span></span>
<span id="cb112-18"><a href="#cb112-18" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(randomprecipMeans <span class="sc">~</span> randomtempMeans, <span class="at">xlim=</span><span class="fu">rev</span>(<span class="fu">range</span>(randomtempMeans, <span class="at">na.rm=</span>T)), <span class="at">pch=</span><span class="dv">16</span>, <span class="at">col=</span><span class="st">"gray"</span>, <span class="at">ylab=</span><span class="st">"Mean annual precipitation (cm)"</span>, <span class="at">xlab=</span><span class="st">"mean annual temperature (°C)"</span>, <span class="at">type=</span><span class="st">"n"</span>, <span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>, <span class="fu">max</span>(<span class="fu">c</span>(<span class="fu">max</span>(randomprecipMeans), <span class="fu">max</span>(sampleprecipMeans, <span class="at">na.rm=</span>T)))))</span>
<span id="cb112-19"><a href="#cb112-19" aria-hidden="true" tabindex="-1"></a>    pointcols<span class="ot">&lt;-</span><span class="fu">rep</span>(<span class="st">"gray"</span>, nrand)</span>
<span id="cb112-20"><a href="#cb112-20" aria-hidden="true" tabindex="-1"></a>    pointcols[<span class="fu">which</span>(randompoints[,<span class="dv">1</span>]<span class="sc">&lt;=</span>temp95)]<span class="ot">&lt;-</span><span class="st">"darkblue"</span></span>
<span id="cb112-21"><a href="#cb112-21" aria-hidden="true" tabindex="-1"></a>    pointcols[<span class="fu">which</span>(randompoints[,<span class="dv">2</span>]<span class="sc">&gt;=</span>precip95)]<span class="ot">&lt;-</span><span class="st">"darkgreen"</span></span>
<span id="cb112-22"><a href="#cb112-22" aria-hidden="true" tabindex="-1"></a>    pointcols[<span class="fu">which</span>(randompoints[, <span class="dv">1</span>] <span class="sc">&lt;=</span> temp95 <span class="sc">&amp;</span> randompoints[, <span class="dv">2</span>] <span class="sc">&gt;=</span> precip95)] <span class="ot">&lt;-</span> <span class="st">"orange"</span></span>
<span id="cb112-23"><a href="#cb112-23" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(randompoints[,<span class="dv">1</span>], randompoints[,<span class="dv">2</span>], <span class="at">pch=</span><span class="dv">16</span>, <span class="at">col=</span> pointcols)</span>
<span id="cb112-24"><a href="#cb112-24" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(<span class="at">x=</span>sampletempMeans, <span class="at">y=</span>sampleprecipMeans, <span class="at">pch=</span><span class="dv">3</span>, <span class="at">col=</span><span class="fu">alpha</span>(<span class="st">"black"</span>, <span class="fl">0.2</span>))</span>
<span id="cb112-25"><a href="#cb112-25" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h=</span>precip95, <span class="at">v=</span>temp95, <span class="at">lty=</span><span class="dv">2</span>, <span class="at">col=</span><span class="st">"black"</span>, <span class="at">lwd=</span><span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="geocov_figs1_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="figure-1a.-map-geography-of-plant-barcodes." class="level1">
<h1>Figure 1A. Map: Geography of plant barcodes.</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a><span class="co"># map for places inside or outside climate envelopes</span></span>
<span id="cb113-2"><a href="#cb113-2" aria-hidden="true" tabindex="-1"></a><span class="co"># make raster by thresholds</span></span>
<span id="cb113-3"><a href="#cb113-3" aria-hidden="true" tabindex="-1"></a>threshtemp <span class="ot">&lt;-</span> t.mean</span>
<span id="cb113-4"><a href="#cb113-4" aria-hidden="true" tabindex="-1"></a><span class="fu">values</span>(threshtemp) <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">values</span>(threshtemp<span class="sc">&lt;</span><span class="fu">as.numeric</span>(temp95)))</span>
<span id="cb113-5"><a href="#cb113-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-6"><a href="#cb113-6" aria-hidden="true" tabindex="-1"></a><span class="fu">quantile</span>(<span class="fu">values</span>(p.mean), <span class="at">na.rm=</span>T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>    0%    25%    50%    75%   100% 
   0.0   11.0   33.8   68.7 1119.1 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb115"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a>threshprecip <span class="ot">&lt;-</span> p.mean</span>
<span id="cb115-2"><a href="#cb115-2" aria-hidden="true" tabindex="-1"></a><span class="fu">values</span>(threshprecip) <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">values</span>(threshprecip<span class="sc">&gt;</span><span class="fu">as.numeric</span>(precip95)))</span>
<span id="cb115-3"><a href="#cb115-3" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(threshprecip)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "RasterLayer"
attr(,"package")
[1] "raster"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb117"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1" aria-hidden="true" tabindex="-1"></a><span class="co"># create ggplot-usable world map </span></span>
<span id="cb117-2"><a href="#cb117-2" aria-hidden="true" tabindex="-1"></a>world <span class="ot">&lt;-</span> sf<span class="sc">::</span><span class="fu">st_as_sf</span>(<span class="fu">map</span>(<span class="st">"world"</span>, <span class="at">plot =</span> <span class="cn">FALSE</span>, <span class="at">fill =</span> <span class="cn">TRUE</span>))</span>
<span id="cb117-3"><a href="#cb117-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-4"><a href="#cb117-4" aria-hidden="true" tabindex="-1"></a><span class="co"># transform all features to longlat and WGS84 - first, world</span></span>
<span id="cb117-5"><a href="#cb117-5" aria-hidden="true" tabindex="-1"></a><span class="fu">st_crs</span>(world)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Coordinate Reference System:
  User input: +proj=longlat +ellps=clrk66 +no_defs +type=crs 
  wkt:
GEOGCRS["unknown",
    DATUM["Unknown based on Clarke 1866 ellipsoid",
        ELLIPSOID["Clarke 1866",6378206.4,294.978698213898,
            LENGTHUNIT["metre",1,
                ID["EPSG",9001]]]],
    PRIMEM["Greenwich",0,
        ANGLEUNIT["degree",0.0174532925199433],
        ID["EPSG",8901]],
    CS[ellipsoidal,2],
        AXIS["longitude",east,
            ORDER[1],
            ANGLEUNIT["degree",0.0174532925199433,
                ID["EPSG",9122]]],
        AXIS["latitude",north,
            ORDER[2],
            ANGLEUNIT["degree",0.0174532925199433,
                ID["EPSG",9122]]]]</code></pre>
</div>
<div class="sourceCode cell-code" id="cb119"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1" aria-hidden="true" tabindex="-1"></a>world <span class="ot">&lt;-</span> <span class="fu">st_transform</span>(world,<span class="fu">CRS</span>(<span class="st">"+proj=longlat +datum=WGS84"</span>)) <span class="co"># the goal here is to choose a projection and datum that every spatial object we add to the map will be converted to before plotting</span></span>
<span id="cb119-2"><a href="#cb119-2" aria-hidden="true" tabindex="-1"></a><span class="fu">st_crs</span>(world) <span class="co"># check coordinate system to ensure projection worked</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Coordinate Reference System:
  User input: GEOGCRS["unknown",
    DATUM["World Geodetic System 1984",
        ELLIPSOID["WGS 84",6378137,298.257223563,
            LENGTHUNIT["metre",1]],
        ID["EPSG",6326]],
    PRIMEM["Greenwich",0,
        ANGLEUNIT["degree",0.0174532925199433],
        ID["EPSG",8901]],
    CS[ellipsoidal,2],
        AXIS["longitude",east,
            ORDER[1],
            ANGLEUNIT["degree",0.0174532925199433,
                ID["EPSG",9122]]],
        AXIS["latitude",north,
            ORDER[2],
            ANGLEUNIT["degree",0.0174532925199433,
                ID["EPSG",9122]]]] 
  wkt:
GEOGCRS["unknown",
    DATUM["World Geodetic System 1984",
        ELLIPSOID["WGS 84",6378137,298.257223563,
            LENGTHUNIT["metre",1]],
        ID["EPSG",6326]],
    PRIMEM["Greenwich",0,
        ANGLEUNIT["degree",0.0174532925199433],
        ID["EPSG",8901]],
    CS[ellipsoidal,2],
        AXIS["longitude",east,
            ORDER[1],
            ANGLEUNIT["degree",0.0174532925199433,
                ID["EPSG",9122]]],
        AXIS["latitude",north,
            ORDER[2],
            ANGLEUNIT["degree",0.0174532925199433,
                ID["EPSG",9122]]]]</code></pre>
</div>
<div class="sourceCode cell-code" id="cb121"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1" aria-hidden="true" tabindex="-1"></a>world <span class="ot">&lt;-</span> <span class="fu">fortify</span>(world) <span class="co"># re-examine features</span></span>
<span id="cb121-2"><a href="#cb121-2" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(world)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "sf"         "data.frame"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb123"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1" aria-hidden="true" tabindex="-1"></a><span class="co"># transform all features to longlat and WGS84 - next, threshtemp</span></span>
<span id="cb123-2"><a href="#cb123-2" aria-hidden="true" tabindex="-1"></a><span class="fu">st_crs</span>(threshtemp)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Coordinate Reference System:
  User input: +proj=longlat +datum=WGS84 +no_defs 
  wkt:
GEOGCRS["unknown",
    DATUM["World Geodetic System 1984",
        ELLIPSOID["WGS 84",6378137,298.257223563,
            LENGTHUNIT["metre",1]],
        ID["EPSG",6326]],
    PRIMEM["Greenwich",0,
        ANGLEUNIT["degree",0.0174532925199433],
        ID["EPSG",8901]],
    CS[ellipsoidal,2],
        AXIS["longitude",east,
            ORDER[1],
            ANGLEUNIT["degree",0.0174532925199433,
                ID["EPSG",9122]]],
        AXIS["latitude",north,
            ORDER[2],
            ANGLEUNIT["degree",0.0174532925199433,
                ID["EPSG",9122]]]]</code></pre>
</div>
<div class="sourceCode cell-code" id="cb125"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1" aria-hidden="true" tabindex="-1"></a>newthreshtemp <span class="ot">&lt;-</span> <span class="st">" +proj=longlat +datum=WGS84"</span></span>
<span id="cb125-2"><a href="#cb125-2" aria-hidden="true" tabindex="-1"></a><span class="co"># simplest approach</span></span>
<span id="cb125-3"><a href="#cb125-3" aria-hidden="true" tabindex="-1"></a>threshtemp <span class="ot">&lt;-</span> <span class="fu">projectRaster</span>(threshtemp, <span class="at">crs=</span>newthreshtemp)</span>
<span id="cb125-4"><a href="#cb125-4" aria-hidden="true" tabindex="-1"></a><span class="fu">st_crs</span>(threshtemp) <span class="co"># check coordinate system to ensure projection worked</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Coordinate Reference System:
  User input: +proj=longlat +datum=WGS84 +no_defs 
  wkt:
GEOGCRS["unknown",
    DATUM["World Geodetic System 1984",
        ELLIPSOID["WGS 84",6378137,298.257223563,
            LENGTHUNIT["metre",1]],
        ID["EPSG",6326]],
    PRIMEM["Greenwich",0,
        ANGLEUNIT["degree",0.0174532925199433],
        ID["EPSG",8901]],
    CS[ellipsoidal,2],
        AXIS["longitude",east,
            ORDER[1],
            ANGLEUNIT["degree",0.0174532925199433,
                ID["EPSG",9122]]],
        AXIS["latitude",north,
            ORDER[2],
            ANGLEUNIT["degree",0.0174532925199433,
                ID["EPSG",9122]]]]</code></pre>
</div>
<div class="sourceCode cell-code" id="cb127"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb127-1"><a href="#cb127-1" aria-hidden="true" tabindex="-1"></a>threshtemp <span class="co">#re-examine features</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>class      : RasterLayer 
dimensions : 1078, 2156, 2324168  (nrow, ncol, ncell)
resolution : 0.167, 0.167  (x, y)
extent     : -180, 180.052, -90.026, 90  (xmin, xmax, ymin, ymax)
crs        : +proj=longlat +datum=WGS84 +no_defs 
source     : memory
names      : layer 
values     : 0, 1  (min, max)</code></pre>
</div>
<div class="sourceCode cell-code" id="cb129"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1" aria-hidden="true" tabindex="-1"></a><span class="co"># transform all features to longlat and WGS84 - next, threshprecip</span></span>
<span id="cb129-2"><a href="#cb129-2" aria-hidden="true" tabindex="-1"></a><span class="fu">st_crs</span>(threshprecip)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Coordinate Reference System:
  User input: +proj=longlat +datum=WGS84 +no_defs 
  wkt:
GEOGCRS["unknown",
    DATUM["World Geodetic System 1984",
        ELLIPSOID["WGS 84",6378137,298.257223563,
            LENGTHUNIT["metre",1]],
        ID["EPSG",6326]],
    PRIMEM["Greenwich",0,
        ANGLEUNIT["degree",0.0174532925199433],
        ID["EPSG",8901]],
    CS[ellipsoidal,2],
        AXIS["longitude",east,
            ORDER[1],
            ANGLEUNIT["degree",0.0174532925199433,
                ID["EPSG",9122]]],
        AXIS["latitude",north,
            ORDER[2],
            ANGLEUNIT["degree",0.0174532925199433,
                ID["EPSG",9122]]]]</code></pre>
</div>
<div class="sourceCode cell-code" id="cb131"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb131-1"><a href="#cb131-1" aria-hidden="true" tabindex="-1"></a>newthreshprecip <span class="ot">&lt;-</span> <span class="st">" +proj=longlat +datum=WGS84"</span></span>
<span id="cb131-2"><a href="#cb131-2" aria-hidden="true" tabindex="-1"></a><span class="co"># simplest approach</span></span>
<span id="cb131-3"><a href="#cb131-3" aria-hidden="true" tabindex="-1"></a>threshprecip <span class="ot">&lt;-</span> <span class="fu">projectRaster</span>(threshprecip, <span class="at">crs=</span>newthreshprecip)</span>
<span id="cb131-4"><a href="#cb131-4" aria-hidden="true" tabindex="-1"></a><span class="fu">st_crs</span>(threshprecip) <span class="co"># check coordinate system to ensure projection worked</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Coordinate Reference System:
  User input: +proj=longlat +datum=WGS84 +no_defs 
  wkt:
GEOGCRS["unknown",
    DATUM["World Geodetic System 1984",
        ELLIPSOID["WGS 84",6378137,298.257223563,
            LENGTHUNIT["metre",1]],
        ID["EPSG",6326]],
    PRIMEM["Greenwich",0,
        ANGLEUNIT["degree",0.0174532925199433],
        ID["EPSG",8901]],
    CS[ellipsoidal,2],
        AXIS["longitude",east,
            ORDER[1],
            ANGLEUNIT["degree",0.0174532925199433,
                ID["EPSG",9122]]],
        AXIS["latitude",north,
            ORDER[2],
            ANGLEUNIT["degree",0.0174532925199433,
                ID["EPSG",9122]]]]</code></pre>
</div>
<div class="sourceCode cell-code" id="cb133"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb133-1"><a href="#cb133-1" aria-hidden="true" tabindex="-1"></a>threshprecip <span class="co"># re-examine features</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>class      : RasterLayer 
dimensions : 1078, 2156, 2324168  (nrow, ncol, ncell)
resolution : 0.167, 0.167  (x, y)
extent     : -180, 180.052, -90.026, 90  (xmin, xmax, ymin, ymax)
crs        : +proj=longlat +datum=WGS84 +no_defs 
source     : memory
names      : layer 
values     : 0, 1  (min, max)</code></pre>
</div>
<div class="sourceCode cell-code" id="cb135"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb135-1"><a href="#cb135-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a data frame for ggplot purposes - threshtemp</span></span>
<span id="cb135-2"><a href="#cb135-2" aria-hidden="true" tabindex="-1"></a>threshtemp_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(threshtemp, <span class="at">xy =</span> <span class="cn">TRUE</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb135-3"><a href="#cb135-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(threshtemp_df, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>             x       y layer
80618 -38.8015 83.7375     0
80619 -38.6345 83.7375     0</code></pre>
</div>
<div class="sourceCode cell-code" id="cb137"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb137-1"><a href="#cb137-1" aria-hidden="true" tabindex="-1"></a>threshtemp_df_subset <span class="ot">&lt;-</span> <span class="fu">subset</span>(threshtemp_df, layer <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb137-2"><a href="#cb137-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-3"><a href="#cb137-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a data frame for ggplot purposes - threshprecip</span></span>
<span id="cb137-4"><a href="#cb137-4" aria-hidden="true" tabindex="-1"></a>threshprecip_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(threshprecip, <span class="at">xy =</span> <span class="cn">TRUE</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb137-5"><a href="#cb137-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(threshprecip_df, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>             x       y layer
80618 -38.8015 83.7375     0
80619 -38.6345 83.7375     0</code></pre>
</div>
<div class="sourceCode cell-code" id="cb139"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb139-1"><a href="#cb139-1" aria-hidden="true" tabindex="-1"></a>threshprecip_df_subset <span class="ot">&lt;-</span> <span class="fu">subset</span>(threshprecip_df, layer <span class="sc">&gt;</span> <span class="dv">0</span> )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb140"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb140-1"><a href="#cb140-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot geography of plant DNA barcodes</span></span>
<span id="cb140-2"><a href="#cb140-2" aria-hidden="true" tabindex="-1"></a><span class="co">#ggplot wrapper of map()</span></span>
<span id="cb140-3"><a href="#cb140-3" aria-hidden="true" tabindex="-1"></a>world_map <span class="ot">&lt;-</span> <span class="fu">map_data</span>(<span class="st">"world"</span>) </span>
<span id="cb140-4"><a href="#cb140-4" aria-hidden="true" tabindex="-1"></a><span class="fu">crs</span>(world_map)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] NA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb142"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb142-1"><a href="#cb142-1" aria-hidden="true" tabindex="-1"></a>ggmap4 <span class="ot">&lt;-</span><span class="fu">ggplot</span>()  <span class="sc">+</span> <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb142-2"><a href="#cb142-2" aria-hidden="true" tabindex="-1"></a> <span class="fu">geom_map</span>(<span class="at">dat =</span> world_map, <span class="at">map =</span> world_map, <span class="fu">aes</span>(<span class="at">map_id =</span> region), <span class="at">fill =</span> <span class="st">"white"</span>, <span class="at">color =</span> <span class="st">"#7f7f7f"</span>, <span class="at">size =</span> <span class="fl">0.25</span>)  <span class="sc">+</span></span>
<span id="cb142-3"><a href="#cb142-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_raster</span>(<span class="at">data =</span> threshtemp_df, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">fill =</span> layer), <span class="at">alpha =</span> <span class="fl">0.9</span>) <span class="sc">+</span> </span>
<span id="cb142-4"><a href="#cb142-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradient</span>(<span class="at">low =</span> <span class="st">"transparent"</span>, <span class="at">high =</span> <span class="st">"darkblue"</span>, <span class="at">aesthetics =</span> <span class="st">'fill'</span>) <span class="sc">+</span> </span>
<span id="cb142-5"><a href="#cb142-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="at">geom=</span><span class="st">"raster"</span>, <span class="at">x=</span>threshprecip_df_subset<span class="sc">$</span>x, <span class="at">y=</span>threshprecip_df_subset<span class="sc">$</span>y, <span class="at">alpha=</span>.<span class="dv">9</span>,</span>
<span id="cb142-6"><a href="#cb142-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">fill =</span> scales<span class="sc">::</span><span class="fu">colour_ramp</span>(<span class="fu">c</span>(<span class="st">"transparent"</span>,<span class="st">"darkgreen"</span>))(threshprecip_df_subset<span class="sc">$</span>layer)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">data =</span> combtab, <span class="fu">aes</span>(<span class="at">x =</span> lon, <span class="at">y =</span> lat), <span class="at">pch =</span> <span class="dv">3</span>, <span class="at">size =</span> <span class="fl">0.05</span>, <span class="at">stroke =</span> <span class="fl">0.25</span>, <span class="at">color =</span> <span class="fu">alpha</span>(<span class="st">"darkgray"</span>, <span class="fl">0.9</span>)) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">""</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">""</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks=</span><span class="fu">seq</span>(<span class="sc">-</span><span class="dv">90</span>,<span class="dv">90</span>,<span class="dv">45</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">90</span>, <span class="dv">90</span>)) <span class="sc">+</span> <span class="fu">scale_x_continuous</span>(<span class="at">breaks=</span><span class="fu">seq</span>(<span class="sc">-</span><span class="dv">180</span>,<span class="dv">180</span>,<span class="dv">60</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">180</span>, <span class="dv">180</span>)) <span class="sc">+</span>  <span class="fu">ylab</span>(<span class="st">""</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">""</span>)<span class="sc">+</span> <span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_blank</span>(), <span class="at">axis.text =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb142-7"><a href="#cb142-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="fl">0.5</span>) <span class="sc">+</span> <span class="co"># Equator line</span></span>
<span id="cb142-8"><a href="#cb142-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fl">23.5</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="fl">0.5</span>) <span class="sc">+</span> <span class="co"># Tropic of Cancer</span></span>
<span id="cb142-9"><a href="#cb142-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fl">23.5</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="fl">0.5</span>)  <span class="co"># Tropic of Capricorn</span></span>
<span id="cb142-10"><a href="#cb142-10" aria-hidden="true" tabindex="-1"></a>ggmap4</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="geocov_figs1_files/figure-html/plot-geocov-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb143"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb143-1"><a href="#cb143-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave("ggmap4_20240626_1.pdf", ggmap4,  width = 13, height = 8.6, unit = "cm")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./building_bold.html" class="pagination-link" aria-label="Building the BOLD dataset">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-title">Building the BOLD dataset</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./geocov_figs2.html" class="pagination-link" aria-label="Coverage by country">
        <span class="nav-page-text"><span class="chapter-title">Coverage by country</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>